=== Content from github.com_cb73b405_20250114_191858.html ===

[Skip to content](#start-of-content)

## Navigation Menu

Toggle navigation

[Sign in](/login?return_to=https%3A%2F%2Fgithub.com%2Ftanghaibao%2Fjcvi%2Fsecurity%2Fadvisories%2FGHSA-x49m-3cw7-gq5q)

* Product

  + [GitHub Copilot
    Write better code with AI](https://github.com/features/copilot)
  + [Security
    Find and fix vulnerabilities](https://github.com/features/security)
  + [Actions
    Automate any workflow](https://github.com/features/actions)
  + [Codespaces
    Instant dev environments](https://github.com/features/codespaces)
  + [Issues
    Plan and track work](https://github.com/features/issues)
  + [Code Review
    Manage code changes](https://github.com/features/code-review)
  + [Discussions
    Collaborate outside of code](https://github.com/features/discussions)
  + [Code Search
    Find more, search less](https://github.com/features/code-search)

  Explore
  + [All features](https://github.com/features)
  + [Documentation](https://docs.github.com)
  + [GitHub Skills](https://skills.github.com)
  + [Blog](https://github.blog)
* Solutions

  By company size
  + [Enterprises](https://github.com/enterprise)
  + [Small and medium teams](https://github.com/team)
  + [Startups](https://github.com/enterprise/startups)
  By use case
  + [DevSecOps](/solutions/use-case/devsecops)
  + [DevOps](/solutions/use-case/devops)
  + [CI/CD](/solutions/use-case/ci-cd)
  + [View all use cases](/solutions/use-case)

  By industry
  + [Healthcare](/solutions/industry/healthcare)
  + [Financial services](/solutions/industry/financial-services)
  + [Manufacturing](/solutions/industry/manufacturing)
  + [Government](/solutions/industry/government)
  + [View all industries](/solutions/industry)

  [View all solutions](/solutions)
* Resources

  Topics
  + [AI](/resources/articles/ai)
  + [DevOps](/resources/articles/devops)
  + [Security](/resources/articles/security)
  + [Software Development](/resources/articles/software-development)
  + [View all](/resources/articles)

  Explore
  + [Learning Pathways](https://resources.github.com/learn/pathways)
  + [White papers, Ebooks, Webinars](https://resources.github.com)
  + [Customer Stories](https://github.com/customer-stories)
  + [Partners](https://partner.github.com)
  + [Executive Insights](https://github.com/solutions/executive-insights)
* Open Source

  + [GitHub Sponsors
    Fund open source developers](/sponsors)
  + [The ReadME Project
    GitHub community articles](https://github.com/readme)
  Repositories
  + [Topics](https://github.com/topics)
  + [Trending](https://github.com/trending)
  + [Collections](https://github.com/collections)
* Enterprise

  + [Enterprise platform
    AI-powered developer platform](/enterprise)
  Available add-ons
  + [Advanced Security
    Enterprise-grade security features](https://github.com/enterprise/advanced-security)
  + [GitHub Copilot
    Enterprise-grade AI features](/features/copilot#enterprise)
  + [Premium Support
    Enterprise-grade 24/7 support](/premium-support)
* [Pricing](https://github.com/pricing)

Search or jump to...

# Search code, repositories, users, issues, pull requests...

Search

Clear

[Search syntax tips](https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax)

# Provide feedback

We read every piece of feedback, and take your input very seriously.

Include my email address so I can be contacted

  Cancel

 Submit feedback

# Saved searches

## Use saved searches to filter your results more quickly

Name

Query

To see all available qualifiers, see our [documentation](https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax).

  Cancel

 Create saved search

[Sign in](/login?return_to=https%3A%2F%2Fgithub.com%2Ftanghaibao%2Fjcvi%2Fsecurity%2Fadvisories%2FGHSA-x49m-3cw7-gq5q)

[Sign up](/signup?ref_cta=Sign+up&ref_loc=header+logged+out&ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E%2Frepos%2Fadvisories%2Fshow&source=header-repo&source_repo=tanghaibao%2Fjcvi)
Reseting focus

You signed in with another tab or window. Reload to refresh your session.
You signed out in another tab or window. Reload to refresh your session.
You switched accounts on another tab or window. Reload to refresh your session.

Dismiss alert

{{ message }}

[tanghaibao](/tanghaibao)
/
**[jcvi](/tanghaibao/jcvi)**
Public

* [Notifications](/login?return_to=%2Ftanghaibao%2Fjcvi) You must be signed in to change notification settings
* [Fork
  189](/login?return_to=%2Ftanghaibao%2Fjcvi)
* [Star
   787](/login?return_to=%2Ftanghaibao%2Fjcvi)

* [Code](/tanghaibao/jcvi)
* [Issues
  49](/tanghaibao/jcvi/issues)
* [Pull requests
  1](/tanghaibao/jcvi/pulls)
* [Discussions](/tanghaibao/jcvi/discussions)
* [Actions](/tanghaibao/jcvi/actions)
* [Wiki](/tanghaibao/jcvi/wiki)
* [Security](/tanghaibao/jcvi/security)
* [Insights](/tanghaibao/jcvi/pulse)

Additional navigation options

* [Code](/tanghaibao/jcvi)
* [Issues](/tanghaibao/jcvi/issues)
* [Pull requests](/tanghaibao/jcvi/pulls)
* [Discussions](/tanghaibao/jcvi/discussions)
* [Actions](/tanghaibao/jcvi/actions)
* [Wiki](/tanghaibao/jcvi/wiki)
* [Security](/tanghaibao/jcvi/security)
* [Insights](/tanghaibao/jcvi/pulse)

# Configuration Injection in tanghaibao/jcvi due to unsanitized user input

High

[tanghaibao](/tanghaibao)
published
GHSA-x49m-3cw7-gq5q
Jun 23, 2023

## Package

pip

jcvi
([pip](/advisories?query=ecosystem%3Apip))

## Affected versions

<=v1.3.5

## Patched versions

None

## Description

### Summary

A configuration injection happens when user input is considered by the application in an unsanitized format and can reach the configuration file. A malicious user may craft a special payload that may lead to a command injection.

### PoC

The vulnerable code snippet is [/jcvi/apps/base.py#LL2227C1-L2228C41](https://github.com/tanghaibao/jcvi/blob/cede6c65c8e7603cb266bc3395ac8f915ea9eac7/jcvi/apps/base.py#LL2227C1-L2228C41). Under some circumstances a user input is retrieved and stored within the `fullpath` variable which reaches the configuration file `~/.jcvirc`.

```
        fullpath = input(msg).strip()
        config.set(PATH, name, fullpath)
```

I ripped a part of the codebase into a runnable PoC as follows. All the PoC does is call the `getpath()` function under some circumstances.

```
from configparser import (
    ConfigParser,
    RawConfigParser,
    NoOptionError,
    NoSectionError,
    ParsingError,
)

import errno
import os
import sys
import os.path as op
import shutil
import signal
import sys
import logging

def is_exe(fpath):
    return op.isfile(fpath) and os.access(fpath, os.X_OK)

def which(program):
    """
    Emulates the unix which command.

    >>> which("cat")
    "/bin/cat"
    >>> which("nosuchprogram")
    """
    fpath, fname = op.split(program)
    if fpath:
        if is_exe(program):
            return program
    else:
        for path in os.environ["PATH"].split(os.pathsep):
            exe_file = op.join(path, program)
            if is_exe(exe_file):
                return exe_file

    return None

def getpath(cmd, name=None, url=None, cfg="~/.jcvirc", warn="exit"):
    """
    Get install locations of common binaries
    First, check ~/.jcvirc file to get the full path
    If not present, ask on the console and store
    """
    p = which(cmd)  # if in PATH, just returns it
    if p:
        return p

    PATH = "Path"
    config = RawConfigParser()
    cfg = op.expanduser(cfg)
    changed = False
    if op.exists(cfg):
        config.read(cfg)

    assert name is not None, "Need a program name"

    try:
        fullpath = config.get(PATH, name)
    except NoSectionError:
        config.add_section(PATH)
        changed = True

    try:
        fullpath = config.get(PATH, name)
    except NoOptionError:
        msg = "=== Configure path for {0} ===\n".format(name, cfg)
        if url:
            msg += "URL: {0}\n".format(url)
        msg += "[Directory that contains `{0}`]: ".format(cmd)
        fullpath = input(msg).strip()
        config.set(PATH, name, fullpath)
        changed = True

    path = op.join(op.expanduser(fullpath), cmd)
    if warn == "exit":
        try:
            assert is_exe(path), "***ERROR: Cannot execute binary `{0}`. ".format(path)
        except AssertionError as e:
            sys.exit("{0!s}Please verify and rerun.".format(e))

    if changed:
        configfile = open(cfg, "w")
        config.write(configfile)
        logging.debug("Configuration written to `{0}`.".format(cfg))

    return path

# Call to getpath
path = getpath("not-part-of-path", name="CLUSTALW2", warn="warn")
print(path)
```

To run the PoC, you need to remove the config file `~/.jcvirc` to emulate the first run,

```
# Run the PoC with the payload
echo -e "e\rvvvvvvvv = zzzzzzzz\n" | python3 poc.py
```

[![image](https://user-images.githubusercontent.com/13036531/247852364-f8a384a3-fc62-41ca-b467-877d197ac6ff.png)](https://user-images.githubusercontent.com/13036531/247852364-f8a384a3-fc62-41ca-b467-877d197ac6ff.png)

You can notice the random key/value characters `vvvvvvvv = zzzzzzzz` were successfully injected.

### Impact

The impact of a configuration injection may vary. Under some conditions, it may lead to command injection if there is for instance shell code execution from the configuration file values.

### Severity

High

### CVE ID

CVE-2023-35932

### Weaknesses

[CWE-77](/advisories?query=cwe%3A77)

### Credits

* [![@Sim4n6](https://avatars.githubusercontent.com/u/13036531?s=40&v=4)](/Sim4n6)
  [Sim4n6](/Sim4n6)
  Analyst

## Footer

© 2025 GitHub, Inc.

### Footer navigation

* [Terms](https://docs.github.com/site-policy/github-terms/github-terms-of-service)
* [Privacy](https://docs.github.com/site-policy/privacy-policies/github-privacy-statement)
* [Security](https://github.com/security)
* [Status](https://www.githubstatus.com/)
* [Docs](https://docs.github.com/)
* [Contact](https://support.github.com?tags=dotcom-footer)
* Manage cookies
* Do not share my personal information

You can’t perform that action at this time.



=== Content from github.com_063da55b_20250114_191856.html ===

[Skip to content](#start-of-content)

## Navigation Menu

Toggle navigation

[Sign in](/login?return_to=https%3A%2F%2Fgithub.com%2Ftanghaibao%2Fjcvi%2Fblob%2Fcede6c65c8e7603cb266bc3395ac8f915ea9eac7%2Fjcvi%2Fapps%2Fbase.py)

* Product

  + [GitHub Copilot
    Write better code with AI](https://github.com/features/copilot)
  + [Security
    Find and fix vulnerabilities](https://github.com/features/security)
  + [Actions
    Automate any workflow](https://github.com/features/actions)
  + [Codespaces
    Instant dev environments](https://github.com/features/codespaces)
  + [Issues
    Plan and track work](https://github.com/features/issues)
  + [Code Review
    Manage code changes](https://github.com/features/code-review)
  + [Discussions
    Collaborate outside of code](https://github.com/features/discussions)
  + [Code Search
    Find more, search less](https://github.com/features/code-search)

  Explore
  + [All features](https://github.com/features)
  + [Documentation](https://docs.github.com)
  + [GitHub Skills](https://skills.github.com)
  + [Blog](https://github.blog)
* Solutions

  By company size
  + [Enterprises](https://github.com/enterprise)
  + [Small and medium teams](https://github.com/team)
  + [Startups](https://github.com/enterprise/startups)
  By use case
  + [DevSecOps](/solutions/use-case/devsecops)
  + [DevOps](/solutions/use-case/devops)
  + [CI/CD](/solutions/use-case/ci-cd)
  + [View all use cases](/solutions/use-case)

  By industry
  + [Healthcare](/solutions/industry/healthcare)
  + [Financial services](/solutions/industry/financial-services)
  + [Manufacturing](/solutions/industry/manufacturing)
  + [Government](/solutions/industry/government)
  + [View all industries](/solutions/industry)

  [View all solutions](/solutions)
* Resources

  Topics
  + [AI](/resources/articles/ai)
  + [DevOps](/resources/articles/devops)
  + [Security](/resources/articles/security)
  + [Software Development](/resources/articles/software-development)
  + [View all](/resources/articles)

  Explore
  + [Learning Pathways](https://resources.github.com/learn/pathways)
  + [White papers, Ebooks, Webinars](https://resources.github.com)
  + [Customer Stories](https://github.com/customer-stories)
  + [Partners](https://partner.github.com)
  + [Executive Insights](https://github.com/solutions/executive-insights)
* Open Source

  + [GitHub Sponsors
    Fund open source developers](/sponsors)
  + [The ReadME Project
    GitHub community articles](https://github.com/readme)
  Repositories
  + [Topics](https://github.com/topics)
  + [Trending](https://github.com/trending)
  + [Collections](https://github.com/collections)
* Enterprise

  + [Enterprise platform
    AI-powered developer platform](/enterprise)
  Available add-ons
  + [Advanced Security
    Enterprise-grade security features](https://github.com/enterprise/advanced-security)
  + [GitHub Copilot
    Enterprise-grade AI features](/features/copilot#enterprise)
  + [Premium Support
    Enterprise-grade 24/7 support](/premium-support)
* [Pricing](https://github.com/pricing)

Search or jump to...

# Search code, repositories, users, issues, pull requests...

Search

Clear

[Search syntax tips](https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax)

# Provide feedback

We read every piece of feedback, and take your input very seriously.

Include my email address so I can be contacted

  Cancel

 Submit feedback

# Saved searches

## Use saved searches to filter your results more quickly

Name

Query

To see all available qualifiers, see our [documentation](https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax).

  Cancel

 Create saved search

[Sign in](/login?return_to=https%3A%2F%2Fgithub.com%2Ftanghaibao%2Fjcvi%2Fblob%2Fcede6c65c8e7603cb266bc3395ac8f915ea9eac7%2Fjcvi%2Fapps%2Fbase.py)

[Sign up](/signup?ref_cta=Sign+up&ref_loc=header+logged+out&ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E%2Fblob%2Fshow&source=header-repo&source_repo=tanghaibao%2Fjcvi)
Reseting focus

You signed in with another tab or window. Reload to refresh your session.
You signed out in another tab or window. Reload to refresh your session.
You switched accounts on another tab or window. Reload to refresh your session.

Dismiss alert

{{ message }}

[tanghaibao](/tanghaibao)
/
**[jcvi](/tanghaibao/jcvi)**
Public

* [Notifications](/login?return_to=%2Ftanghaibao%2Fjcvi) You must be signed in to change notification settings
* [Fork
  189](/login?return_to=%2Ftanghaibao%2Fjcvi)
* [Star
   787](/login?return_to=%2Ftanghaibao%2Fjcvi)

* [Code](/tanghaibao/jcvi)
* [Issues
  49](/tanghaibao/jcvi/issues)
* [Pull requests
  1](/tanghaibao/jcvi/pulls)
* [Discussions](/tanghaibao/jcvi/discussions)
* [Actions](/tanghaibao/jcvi/actions)
* [Wiki](/tanghaibao/jcvi/wiki)
* [Security](/tanghaibao/jcvi/security)
* [Insights](/tanghaibao/jcvi/pulse)

Additional navigation options

* [Code](/tanghaibao/jcvi)
* [Issues](/tanghaibao/jcvi/issues)
* [Pull requests](/tanghaibao/jcvi/pulls)
* [Discussions](/tanghaibao/jcvi/discussions)
* [Actions](/tanghaibao/jcvi/actions)
* [Wiki](/tanghaibao/jcvi/wiki)
* [Security](/tanghaibao/jcvi/security)
* [Insights](/tanghaibao/jcvi/pulse)

## Files

 cede6c6
## Breadcrumbs

1. [jcvi](/tanghaibao/jcvi/tree/cede6c65c8e7603cb266bc3395ac8f915ea9eac7)
2. /[jcvi](/tanghaibao/jcvi/tree/cede6c65c8e7603cb266bc3395ac8f915ea9eac7/jcvi)
3. /[apps](/tanghaibao/jcvi/tree/cede6c65c8e7603cb266bc3395ac8f915ea9eac7/jcvi/apps)
/
# base.py

Copy path Blame  Blame
## Latest commit

## History

[History](/tanghaibao/jcvi/commits/cede6c65c8e7603cb266bc3395ac8f915ea9eac7/jcvi/apps/base.py)2290 lines (1954 loc) · 66.9 KB cede6c6
## Breadcrumbs

1. [jcvi](/tanghaibao/jcvi/tree/cede6c65c8e7603cb266bc3395ac8f915ea9eac7)
2. /[jcvi](/tanghaibao/jcvi/tree/cede6c65c8e7603cb266bc3395ac8f915ea9eac7/jcvi)
3. /[apps](/tanghaibao/jcvi/tree/cede6c65c8e7603cb266bc3395ac8f915ea9eac7/jcvi/apps)
/
# base.py

Top
## File metadata and controls

* Code
* Blame

2290 lines (1954 loc) · 66.9 KB[Raw](https://github.com/tanghaibao/jcvi/raw/cede6c65c8e7603cb266bc3395ac8f915ea9eac7/jcvi/apps/base.py)1234567891011121314151617181920212223242526272829303132333435363738394041424344454647484950515253545556575859606162636465666768697071727374757677787980818283848586878889909192939495969798991001011021031041051061071081091101111121131141151161171181191201211221231241251261271281291301311321331341351361371381391401411421431441451461471481491501511521531541551561571581591601611621631641651661671681691701711721731741751761771781791801811821831841851861871881891901911921931941951961971981992002012022032042052062072082092102112122132142152162172182192202212222232242252262272282292302312322332342352362372382392402412422432442452462472482492502512522532542552562572582592602612622632642652662672682692702712722732742752762772782792802812822832842852862872882892902912922932942952962972982993003013023033043053063073083093103113123133143153163173183193203213223233243253263273283293303313323333343353363373383393403413423433443453463473483493503513523533543553563573583593603613623633643653663673683693703713723733743753763773783793803813823833843853863873883893903913923933943953963973983994004014024034044054064074084094104114124134144154164174184194204214224234244254264274284294304314324334344354364374384394404414424434444454464474484494504514524534544554564574584594604614624634644654664674684694704714724734744754764774784794804814824834844854864874884894904914924934944954964974984995005015025035045055065075085095105115125135145155165175185195205215225235245255265275285295305315325335345355365375385395405415425435445455465475485495505515525535545555565575585595605615625635645655665675685695705715725735745755765775785795805815825835845855865875885895905915925935945955965975985996006016026036046056066076086096106116126136146156166176186196206216226236246256266276286296306316326336346356366376386396406416426436446456466476486496506516526536546556566576586596606616626636646656666676686696706716726736746756766776786796806816826836846856866876886896906916926936946956966976986997007017027037047057067077087097107117127137147157167177187197207217227237247257267277287297307317327337347357367377387397407417427437447457467477487497507517527537547557567577587597607617627637647657667677687697707717727737747757767777787797807817827837847857867877887897907917927937947957967977987998008018028038048058068078088098108118128138148158168178188198208218228238248258268278288298308318328338348358368378388398408418428438448458468478488498508518528538548558568578588598608618628638648658668678688698708718728738748758768778788798808818828838848858868878888898908918928938948958968978988999009019029039049059069079089099109119129139149159169179189199209219229239249259269279289299309319329339349359369379389399409419429439449459469479489499509519529539549559569579589599609619629639649659669679689699709719729739749759769779789799809819829839849859869879889899909919929939949959969979989991000"""Basic support for running library as script"""import errnoimport osimport timeimport os.path as opimport shutilimport signalimport sysimport loggingimport fnmatchimport functools
from collections.abc import Iterablefrom http.client import HTTPSConnectionfrom urllib.parse import urlencodefrom configparser import ( ConfigParser, RawConfigParser, NoOptionError, NoSectionError, ParsingError,)from socket import gethostnamefrom subprocess import PIPE, call, check\_outputfrom optparse import OptionParser as OptionP, OptionGroup, SUPPRESS\_HELPfrom typing import Any, Collection, List, Optional, Union
from natsort import natsortedfrom rich.logging import Console, RichHandler
from jcvi import \_\_copyright\_\_, \_\_version\_\_
# http://newbebweb.blogspot.com/2012/02/python-head-ioerror-errno-32-broken.htmlnobreakbuffer = lambda: signal.signal(signal.SIGPIPE, signal.SIG\_DFL)nobreakbuffer()os.environ["LC\_ALL"] = "C"JCVIHELP = "JCVI utility libraries {} [{}]\n".format(\_\_version\_\_, \_\_copyright\_\_)
def patch\_debug(func): @functools.wraps(func) def wraps(\*args, \*\*kwargs): import inspect callerframerecord = inspect.stack()[1] frame = callerframerecord[0] info = inspect.getframeinfo(frame) # get caller function name caller = frame.f\_code.co\_name patch\_message = f"{info.filename}:{info.lineno}:{caller}" old\_debug = logging.debug def my\_debug(message: str, \*args, \*\*kwargs): old\_debug(f'{patch\_message} {message}', \*args, \*\*kwargs) logging.debug = my\_debug ret = func(\*args, \*\*kwargs) logging.debug = old\_debug return ret
 return wraps
class ActionDispatcher(object): """ This class will be invoked a) when the base package is run via \_\_main\_\_, listing all MODULESs a) when a directory is run via \_\_main\_\_, listing all SCRIPTs b) when a script is run directly, listing all ACTIONs
 This is controlled through the meta variable, which is automatically determined in get\_meta(). """
 def \_\_init\_\_(self, actions): self.actions = actions if not actions: actions = [(None, None)] self.valid\_actions, self.action\_helps = zip(\*actions)
 def get\_meta(self): args = splitall(sys.argv[0])[-3:] args[-1] = args[-1].replace(".py", "") if args[-2] == "jcvi": meta = "MODULE" elif args[-1] == "\_\_main\_\_": meta = "SCRIPT" else: meta = "ACTION" return meta, args
 def print\_help(self): meta, args = self.get\_meta() if meta == "MODULE": del args[0] args[-1] = meta elif meta == "SCRIPT": args[-1] = meta else: args[-1] += " " + meta
 help = "Usage:\n python -m {0}\n\n\n".format(".".join(args)) help += "Available {0}s:\n".format(meta) max\_action\_len = max(len(action) for action, ah in self.actions) for action, action\_help in sorted(self.actions): action = action.rjust(max\_action\_len + 4) help += ( " | ".join((action, action\_help[0].upper() + action\_help[1:])) + "\n" ) help += "\n" + JCVIHELP
 sys.stderr.write(help) sys.exit(1)
 def dispatch(self, globals): from difflib import get\_close\_matches
 meta = "ACTION" # function is only invoked for listing ACTIONs if len(sys.argv) == 1: self.print\_help()
 action = sys.argv[1]
 if not action in self.valid\_actions: print("[error] {0} not a valid {1}\n".format(action, meta), file=sys.stderr) alt = get\_close\_matches(action, self.valid\_actions) print( "Did you mean one of these?\n\t{0}\n".format(", ".join(alt)), file=sys.stderr, ) self.print\_help()
 globals[action](sys.argv[2:])
class OptionParser(OptionP): def \_\_init\_\_(self, doc): OptionP.\_\_init\_\_(self, doc, epilog=JCVIHELP)
 def parse\_args(self, args=None): dests = set() ol = [] for g in [self] + self.option\_groups: ol += g.option\_list for o in ol: if o.dest in dests: continue self.add\_help\_from\_choices(o) dests.add(o.dest)
 return OptionP.parse\_args(self, args)
 def add\_help\_from\_choices(self, o): if o.help == SUPPRESS\_HELP: return
 default\_tag = "%default" assert o.help, "Option {0} do not have help string".format(o) help\_pf = o.help[:1].upper() + o.help[1:] if "[" in help\_pf: help\_pf = help\_pf.rsplit("[", 1)[0] help\_pf = help\_pf.strip()
 if o.type == "choice": if o.default is None: default\_tag = "guess" ctext = "|".join(natsorted(str(x) for x in o.choices)) if len(ctext) > 100: ctext = ctext[:100] + " ... " choice\_text = "must be one of {0}".format(ctext) o.help = "{0}, {1} [default: {2}]".format(help\_pf, choice\_text, default\_tag) else: o.help = help\_pf if o.default is None: default\_tag = "disabled" if ( o.get\_opt\_string() not in ("--help", "--version") and o.action != "store\_false" ): o.help += " [default: {0}]".format(default\_tag)
 def set\_grid(self): """ Add --grid options for command line programs """ self.add\_option( "--grid", dest="grid", default=False, action="store\_true", help="Run on the grid", )
 def set\_grid\_opts(self, array=False): group = OptionGroup(self, "Grid parameters") group.add\_option( "-l", dest="queue", help="Name of the queue", ) group.add\_option( "-t", dest="threaded", default=None, type="int", help="Append '-pe threaded N'", ) if array: group.add\_option( "-c", dest="concurrency", type="int", help="Append task concurrency limit '-tc N'", ) group.add\_option( "-d", dest="outdir", default=".", help="Specify directory to store grid output/error files", ) group.add\_option( "-N", dest="name", default=None, help="Specify descriptive name for the job" ) group.add\_option( "-H", dest="hold\_jid", default=None, help="Define the job dependency list" ) self.add\_option\_group(group)
 def set\_table(self, sep=",", align=False): group = OptionGroup(self, "Table formatting") group.add\_option("--sep", default=sep, help="Separator") if align: group.add\_option( "--noalign", dest="align", default=True, action="store\_false", help="Cell alignment", ) else: group.add\_option( "--align", default=False, action="store\_true", help="Cell alignment" ) self.add\_option\_group(group)
 def set\_downloader(self, downloader=None): """ Add --downloader options for given command line program. """ from jcvi.utils.ez\_setup import ALL\_DOWNLOADERS
 downloader\_choices = [x[0] for x in ALL\_DOWNLOADERS] self.add\_option( "--downloader", default=downloader, choices=downloader\_choices, help="Use the specified downloader to retrieve resources", )
 def set\_params(self, prog=None, params=""): """ Add --params options for given command line programs """ dest\_prog = "to {0}".format(prog) if prog else "" self.add\_option( "--params", dest="extra", default=params, help="Extra parameters to pass {0}".format(dest\_prog) + " (these WILL NOT be validated)", )
 def set\_outfile(self, outfile="stdout"): """ Add --outfile options to print out to filename. """ self.add\_option("-o", "--outfile", default=outfile, help="Outfile name")
 def set\_outdir(self, outdir="."): self.add\_option("--outdir", default=outdir, help="Specify output directory")
 def set\_email(self): """ Add --email option to specify an email address """ self.add\_option( "--email", default=get\_email\_address(), help="Specify an email address", )
 def set\_tmpdir(self, tmpdir=None): """ Add --temporary\_directory option to specify unix `sort` tmpdir """ self.add\_option( "-T", "--tmpdir", default=tmpdir, help="Use temp directory instead of $TMP" )
 def set\_cpus(self, cpus=0): """ Add --cpus options to specify how many threads to use. """ from multiprocessing import cpu\_count
 max\_cpus = cpu\_count() if not 0 < cpus < max\_cpus: cpus = max\_cpus self.add\_option( "--cpus", default=cpus, type="int", help="Number of CPUs to use, 0=unlimited", )
 def set\_db\_opts(self, dbname="mta4", credentials=True): """ Add db connection specific attributes """ from jcvi.utils.db import valid\_dbconn, get\_profile
 self.add\_option( "--db", default=dbname, dest="dbname", help="Specify name of database to query", ) self.add\_option( "--connector", default="Sybase", dest="dbconn", choices=valid\_dbconn.keys(), help="Specify database connector", ) hostname, username, password = get\_profile() if credentials: self.add\_option("--hostname", default=hostname, help="Specify hostname") self.add\_option( "--username", default=username, help="Username to connect to database" ) self.add\_option( "--password", default=password, help="Password to connect to database" ) self.add\_option("--port", type="int", help="Specify port number")
 def set\_aws\_opts(self, store="hli-mv-data-science/htang"): from jcvi.utils.aws import s3ify
 store = s3ify(store) group = OptionGroup(self, "AWS and Docker options") self.add\_option\_group(group) # https://github.com/hlids/infrastructure/wiki/Docker-calling-convention group.add\_option("--sample\_id", help="Sample ID") group.add\_option("--workflow\_execution\_id", help="Workflow execution ID") group.add\_option("--input\_bam\_path", help="Input BAM location (s3 ok)") group.add\_option("--output\_path", default=store, help="Output s3 path") group.add\_option("--workdir", default=os.getcwd(), help="Specify work dir") group.add\_option( "--nocleanup", default=False, action="store\_true", help="Don't clean up after done", )
 def set\_stripnames(self, default=True): if default: self.add\_option( "--no\_strip\_names", dest="strip\_names", action="store\_false", default=True, help="do not strip alternative splicing " "(e.g. At5g06540.1 -> At5g06540)", ) else: self.add\_option( "--strip\_names", action="store\_true", default=False, help="strip alternative splicing (e.g. At5g06540.1 -> At5g06540)", )
 def set\_fixchrnames(self, orgn="medicago"): self.add\_option( "--fixchrname", default=orgn, dest="fix\_chr\_name", help="Fix quirky chromosome names", )
 def set\_SO\_opts(self): verifySO\_choices = ("verify", "resolve:prefix", "resolve:suffix") self.add\_option( "--verifySO", choices=verifySO\_choices, help="Verify validity of GFF3 feature type against the SO; " + "`resolve` will try to converge towards a valid SO " + "term by removing elements from the feature type " + "string by splitting at underscores. Example: " + "`mRNA\_TE\_gene` resolves to `mRNA` using 'resolve:prefix'", )
 def set\_beds(self): self.add\_option("--qbed", help="Path to qbed") self.add\_option("--sbed", help="Path to sbed")
 def set\_histogram(self, vmin=0, vmax=None, bins=20, xlabel="value", title=None): self.add\_option( "--vmin", default=vmin, type="int", help="Minimum value, inclusive" ) self.add\_option( "--vmax", default=vmax, type="int", help="Maximum value, inclusive" ) self.add\_option( "--bins", default=bins, type="int", help="Number of bins to plot in the histogram", ) self.add\_option("--xlabel", default=xlabel, help="Label on the X-axis") self.add\_option("--title", default=title, help="Title of the plot")
 def set\_sam\_options(self, extra=True, bowtie=False): self.add\_option( "--sam", dest="bam", default=True, action="store\_false", help="Write to SAM file instead of BAM", ) self.add\_option( "--uniq", default=False, action="store\_true", help="Keep only uniquely mapped", ) if bowtie: self.add\_option( "--mapped", default=False, action="store\_true", help="Keep mapped reads" ) self.add\_option( "--unmapped", default=False, action="store\_true", help="Keep unmapped reads" ) if extra: self.set\_cpus() self.set\_params()
 def set\_mingap(self, default=100): self.add\_option( "--mingap", default=default, type="int", help="Minimum size of gaps" )
 def set\_align( self, pctid=None, hitlen=None, pctcov=None, evalue=None, compreh\_pctid=None, compreh\_pctcov=None, intron=None, bpsplice=None, ): if pctid is not None: self.add\_option( "--pctid", default=pctid, type="float", help="Sequence percent identity" ) if hitlen is not None: self.add\_option( "--hitlen", default=hitlen, type="int", help="Minimum overlap length" ) if pctcov is not None: self.add\_option( "--pctcov", default=pctcov, type="int", help="Percentage coverage cutoff", ) if evalue is not None: self.add\_option( "--evalue", default=evalue, type="float", help="E-value cutoff" ) if compreh\_pctid is not None: self.add\_option( "--compreh\_pctid", default=compreh\_pctid, type="int", help="Sequence percent identity cutoff used to " + "build PASA comprehensive transcriptome", ) if compreh\_pctcov is not None: self.add\_option( "--compreh\_pctcov", default=compreh\_pctcov, type="int", help="Percent coverage cutoff used to " + "build PASA comprehensive transcriptome", ) if intron is not None: self.add\_option( "--intron", default=intron, type="int", help="Maximum intron length used for mapping", ) if bpsplice is not None: self.add\_option( "--bpsplice", default=bpsplice, type="int", help="Number of bp of perfect splice boundary", )
 def set\_image\_options( self, args=None, figsize="6x6", dpi=300, format="pdf", font="Helvetica", style="darkgrid", cmap="jet", seed: Optional[int] = None, ): """ Add image format options for given command line programs. """ from jcvi.graphics.base import ( GRAPHIC\_FORMATS, ImageOptions, is\_tex\_available, setup\_theme, )
 allowed\_fonts = ("Helvetica", "Palatino", "Schoolbook", "Arial") allowed\_styles = ("darkgrid", "whitegrid", "dark", "white", "ticks") allowed\_diverge = ( "BrBG", "PiYG", "PRGn", "PuOr", "RdBu", "RdGy", "RdYlBu", "RdYlGn", "Spectral", )
 group = OptionGroup(self, "Image options") self.add\_option\_group(group)
 group.add\_option( "--figsize", default=figsize, help="Figure size `width`x`height` in inches" ) group.add\_option( "--dpi", default=dpi, type="int", help="Physical dot density (dots per inch)", ) group.add\_option( "--format", default=format, choices=GRAPHIC\_FORMATS, help="Generate image of format", ) group.add\_option( "--font", default=font, choices=allowed\_fonts, help="Font name" ) group.add\_option( "--style", default=style, choices=allowed\_styles, help="Axes background" ) group.add\_option( "--diverge", default="PiYG", choices=allowed\_diverge, help="Contrasting color scheme", ) group.add\_option("--cmap", default=cmap, help="Use this color map") group.add\_option( "--notex", default=False, action="store\_true", help="Do not use tex" ) # https://github.com/tanghaibao/jcvi/issues/515#issuecomment-1327305211 if not group.has\_option("--seed"): group.add\_option( "--seed", default=seed, type="int", help="Random seed when assigning colors (supported only for some plots)", )
 if args is None: args = sys.argv[1:]
 opts, args = self.parse\_args(args)
 assert opts.dpi > 0 assert "x" in opts.figsize
 iopts = ImageOptions(opts)
 if opts.notex: logging.info("--notex={}. latex use is disabled.".format(opts.notex)) elif not is\_tex\_available(): if not bool(which("latex")): logging.info("`latex` not found. latex use is disabled.") if not bool(which("lp")): logging.info("`lp` not found. latex use is disabled.")
 setup\_theme(style=opts.style, font=opts.font, usetex=iopts.usetex)
 return opts, args, iopts
 def set\_dotplot\_opts(self, theme: int = 2) -> OptionGroup: """Used in compara.catalog and graphics.dotplot""" from jcvi.graphics.base import set1
 group = OptionGroup(self, "Dot plot parameters") group.add\_option( "--skipempty", default=False, action="store\_true", help="Skip seqids that do not have matches", ) group.add\_option( "--nochpf", default=False, action="store\_true", help="Do not change the contig name", ) group.add\_option( "--nostdpf", default=False, action="store\_true", help="Do not standardize contig names", ) group.add\_option( "--genomenames", type="string", default=None, help="genome names for labeling axes in the form of qname\_sname, " 'eg. "\*Vitis vinifera\*\_\*Oryza sativa\*"', ) group.add\_option( "--theme", choices=[str(x) for x in range(len(set1))], default=str(theme), help="Color index within the palette for contig grid boundaries. Palette contains: {}".format( "|".join(set1) ), ) self.add\_option\_group(group) return group
 def set\_depth(self, depth=50): self.add\_option("--depth", default=depth, type="float", help="Desired depth")
 def set\_rclip(self, rclip=0): self.add\_option( "--rclip", default=rclip, type="int", help="Pair ID is derived from rstrip N chars", )
 def set\_chr(self, chr=",".join([str(x) for x in range(1, 23)] + ["X", "Y", "MT"])): self.add\_option("--chr", default=chr, help="Chromosomes to process")
 def set\_ref(self, ref="/mnt/ref"): self.add\_option("--ref", default=ref, help="Reference folder")
 def set\_cutoff(self, cutoff=0): self.add\_option( "--cutoff", default=cutoff, type="int", help="Distance to call valid links between mates", )
 def set\_mateorientation(self, mateorientation=None): self.add\_option( "--mateorientation", default=mateorientation, choices=("++", "--", "+-", "-+"), help="Use only certain mate orientations", )
 def set\_mates(self, rclip=0, cutoff=0, mateorientation=None): self.set\_rclip(rclip=rclip) self.set\_cutoff(cutoff=cutoff) self.set\_mateorientation(mateorientation=mateorientation)
 def set\_bedpe(self): self.add\_option( "--norc", dest="rc", default=True, action="store\_false", help="Do not reverse complement, expect innie reads", ) self.add\_option( "--minlen", default=2000, type="int", help="Minimum insert size" ) self.add\_option( "--maxlen", default=8000, type="int", help="Maximum insert size" ) self.add\_option( "--dup", default=10, type="int", help="Filter duplicates with coordinates within this distance", )
 def set\_fastq\_names(self): self.add\_option( "--names", default="\*.fq,\*.fastq,\*.fq.gz,\*.fastq.gz", help="File names to search, use comma to separate multiple", )
 def set\_pairs(self): """ %prog pairs <blastfile|samfile|bedfile>
 Report how many paired ends mapped, avg distance between paired ends, etc. Paired reads must have the same prefix, use --rclip to remove trailing part, e.g. /1, /2, or .f, .r, default behavior is to truncate until last char. """ self.set\_usage(self.set\_pairs.\_\_doc\_\_)
 self.add\_option( "--pairsfile", default=None, help="Write valid pairs to pairsfile" ) self.add\_option( "--nrows", default=200000, type="int", help="Only use the first n lines" ) self.set\_mates() self.add\_option( "--pdf", default=False, action="store\_true", help="Print PDF instead ASCII histogram", ) self.add\_option( "--bins", default=20, type="int", help="Number of bins in the histogram" ) self.add\_option( "--distmode", default="ss", choices=("ss", "ee"), help="Distance mode between paired reads, ss is outer distance, " "ee is inner distance", )
 def set\_sep(self, sep="\t", help="Separator in the tabfile", multiple=False): if multiple: help += ", multiple values allowed" self.add\_option("--sep", default=sep, help=help)
 def set\_firstN(self, firstN=100000): self.add\_option( "--firstN", default=firstN, type="int", help="Use only the first N reads" )
 def set\_tag(self, tag=False, specify\_tag=False): if not specify\_tag: self.add\_option( "--tag", default=tag, action="store\_true", help="Add tag (/1, /2) to the read name", ) else: tag\_choices = ["/1", "/2"] self.add\_option( "--tag", default=None, choices=tag\_choices, help="Specify tag to be added to read name", )
 def set\_phred(self, phred=None): phdchoices = ("33", "64") self.add\_option( "--phred", default=phred, choices=phdchoices, help="Phred score offset {0} [default: guess]".format(phdchoices), )
 def set\_size(self, size=0): self.add\_option( "--size", default=size, type="int", help="Insert mean size, stdev assumed to be 20% around mean", )
 def set\_trinity\_opts(self): self.set\_home("trinity") self.set\_home("hpcgridrunner") self.set\_cpus() self.set\_params(prog="Trinity") topts = OptionGroup(self, "General Trinity options") self.add\_option\_group(topts) topts.add\_option( "--max\_memory", default="128G", type="str", help="Jellyfish memory allocation", ) topts.add\_option( "--min\_contig\_length", default=90, type="int", help="Minimum assembled contig length to report", ) topts.add\_option( "--bflyGCThreads", default=None, type="int", help="Threads for garbage collection", ) topts.add\_option( "--grid\_conf\_file", default="JCVI\_SGE.0689.conf", type="str", help="HpcGridRunner config file for supported compute farms", ) topts.add\_option( "--cleanup", default=False, action="store\_true", help="Force clean-up of unwanted files after Trinity run is complete", ) ggopts = OptionGroup(self, "Genome-guided Trinity options") self.add\_option\_group(ggopts) ggopts.add\_option( "--bam", default=None, type="str", help="provide coord-sorted bam file as starting point", ) ggopts.add\_option( "--max\_intron", default=15000, type="int", help="maximum allowed intron length", )
 def set\_pasa\_opts(self, action="assemble"): self.set\_home("pasa") if action == "assemble": self.set\_home("tgi") self.add\_option( "--clean", default=False, action="store\_true", help="Clean transcripts using tgi seqclean", ) self.set\_align(pctid=95, pctcov=90, intron=15000, bpsplice=3) self.add\_option( "--aligners", default="blat,gmap", help="Specify splice aligners to use for mapping", ) self.add\_option( "--fl\_accs", default=None, type="str", help="File containing list of FL-cDNA accessions", ) self.set\_cpus() self.add\_option( "--compreh", default=False, action="store\_true", help="Run comprehensive transcriptome assembly", ) self.set\_align(compreh\_pctid=95, compreh\_pctcov=30) self.add\_option( "--prefix", default="compreh\_init\_build", type="str", help="Prefix for compreh\_trans output file names", ) elif action == "compare": self.add\_option( "--annots\_gff3", default=None, type="str", help="Reference annotation to load and compare against", ) genetic\_code = [ "universal", "Euplotes", "Tetrahymena", "Candida", "Acetabularia", ] self.add\_option( "--genetic\_code", default="universal", choices=genetic\_code, help="Choose translation table", ) self.add\_option( "--pctovl", default=50, type="int", help="Minimum pct overlap between gene and FL assembly", ) self.add\_option( "--pct\_coding", default=50, type="int", help="Minimum pct of cDNA sequence to be protein coding", ) self.add\_option( "--orf\_size", default=0, type="int", help="Minimum size of ORF encoded protein", ) self.add\_option( "--utr\_exons", default=2, type="int", help="Maximum number of UTR exons" ) self.add\_option( "--pctlen\_FL", default=70, type="int", help="Minimum protein length for comparisons involving " + "FL assemblies", ) self.add\_option( "--pctlen\_nonFL", default=70, type="int", help="Minimum protein length for comparisons involving " + "non-FL assemblies", ) self.add\_option( "--pctid\_prot", default=70, type="int", help="Minimum pctid allowed for protein pairwise comparison", ) self.add\_option( "--pct\_aln", default=70, type="int", help="Minimum pct of shorter protein length aligning to " + "update protein or isoform", ) self.add\_option( "--pctovl\_gene", default=80, type="int", help="Minimum pct overlap among genome span of the ORF of " + "each overlapping gene to allow merging", ) self.add\_option( "--stompovl", default="", action="store\_true", help="Ignore alignment results, only consider genome span of ORF", ) self.add\_option( "--trust\_FL", default="", action="store\_true", help="Trust FL-status of cDNA", )
 def set\_annot\_reformat\_opts(self): self.add\_option( "--pad0", default=6, type="int", help="Pad gene identifiers with 0" ) self.add\_option("--prefix", default="Medtr", help="Genome prefix") self.add\_option( "--uc", default=False, action="store\_true", help="Toggle gene identifier upper case", )
 def set\_home(self, prog, default=None): tag = "--{0}\_home".format(prog) default = default or { "amos": "~/code/amos-code", "trinity": "~/export/trinityrnaseq-2.0.6", "hpcgridrunner": "~/export/hpcgridrunner-1.0.2", "cdhit": "~/export/cd-hit-v4.6.1-2012-08-27", "maker": "~/export/maker", "augustus": "~/export/maker/exe/augustus", "pasa": "~/export/PASApipeline-2.0.2", "gatk": "~/export", "gmes": "~/export/gmes", "gt": "~/export/genometools",[View remainder of file in raw view](https://github.com/tanghaibao/jcvi/raw/cede6c65c8e7603cb266bc3395ac8f915ea9eac7/jcvi/apps/base.py)

## Footer

© 2025 GitHub, Inc.

### Footer navigation

* [Terms](https://docs.github.com/site-policy/github-terms/github-terms-of-service)
* [Privacy](https://docs.github.com/site-policy/privacy-policies/github-privacy-statement)
* [Security](https://github.com/security)
* [Status](https://www.githubstatus.com/)
* [Docs](https://docs.github.com/)
* [Contact](https://support.github.com?tags=dotcom-footer)
* Manage cookies
* Do not share my personal information

You can’t perform that action at this time.


