=== Content from launchpad.net_ccc49335_20250121_022618.html ===

[Log in / Register](https://launchpad.net/ubuntu/%2Bsource/bwa/0.7.17-1ubuntu0.1/%2Blogin)

[![](https://launchpadlibrarian.net/606381979/CoF%2064px.png)](https://launchpad.net/ubuntu)

## [Ubuntu](https://launchpad.net/ubuntu)[bwa package](https://launchpad.net/ubuntu/%2Bsource/bwa)

* [Overview](https://launchpad.net/ubuntu/%2Bsource/bwa)
* [Code](https://code.launchpad.net/ubuntu/%2Bsource/bwa)
* [Bugs](https://bugs.launchpad.net/ubuntu/%2Bsource/bwa)
* Blueprints
* [Translations](https://translations.launchpad.net/ubuntu/%2Bsource/bwa)
* [Answers](https://answers.launchpad.net/ubuntu/%2Bsource/bwa)

# bwa 0.7.17-1ubuntu0.1 source package in Ubuntu

## Changelog

```
bwa (0.7.17-1ubuntu0.1) bionic-security; urgency=medium

  * SECURITY UPDATE: Stack buffer overflow via a long sequence name in a .alt
    file
    - debian/patches/CVE-2019-10269.patch: exit if .alt is malformatted
    - CVE-2019-10269

 -- Mike Salvatore <email address hidden>  Mon, 05 Aug 2019 11:36:30 -0400
```

## Upload details

Uploaded by:
[Mike Salvatore](/~mikesalvatore)
on 2019-08-05

Uploaded to:
[Bionic](/ubuntu/bionic)

Original maintainer:
[Ubuntu Developers](/~ubuntu-devel-discuss-lists)

Architectures:
amd64 kfreebsd-amd64

Section:
science

Urgency:
Medium Urgency

## [See full publishing history](%2Bpublishinghistory) Publishing

| Series | Pocket | Published | Component | Section |
| --- | --- | --- | --- | --- |
| [Bionic](/ubuntu/bionic) | updates | on 2019-08-06 | universe | science |
| [Bionic](/ubuntu/bionic) | security | on 2019-08-06 | universe | science |

## Builds

[Bionic](/ubuntu/bionic):
![[FULLYBUILT]](/@@/build-success "Successfully built")
[amd64](/~ubuntu-security-proposed/%2Barchive/ubuntu/ppa/%2Bbuild/17374469)

## Downloads

| File | Size | SHA-256 Checksum |
| --- | --- | --- |
| [bwa\_0.7.17.orig.tar.gz](https://launchpad.net/ubuntu/%2Barchive/primary/%2Bsourcefiles/bwa/0.7.17-1ubuntu0.1/bwa_0.7.17.orig.tar.gz) | 227.1 KiB | 980b9591b61c60042c4a39b9e31ccaad8d17ff179d44d347997825da3fdf47fd |
| [bwa\_0.7.17-1ubuntu0.1.debian.tar.xz](https://launchpad.net/ubuntu/%2Barchive/primary/%2Bsourcefiles/bwa/0.7.17-1ubuntu0.1/bwa_0.7.17-1ubuntu0.1.debian.tar.xz) | 8.6 KiB | 7ba235bdf335524ea3c4491a5887790f6859c061efea0abbca41fe4c8052b7fa |
| [bwa\_0.7.17-1ubuntu0.1.dsc](https://launchpad.net/ubuntu/%2Barchive/primary/%2Bsourcefiles/bwa/0.7.17-1ubuntu0.1/bwa_0.7.17-1ubuntu0.1.dsc) | 2.2 KiB | 578442dbab1deb1f60fb5f8c7b1f9f86e303ceea2db25d2607c3dd6381d008a6 |

### Available diffs

* [diff from 0.7.17-1 (in Debian) to 0.7.17-1ubuntu0.1](http://launchpadlibrarian.net/436104808/bwa_0.7.17-1_0.7.17-1ubuntu0.1.diff.gz) (1.2 KiB)

[View changes file](http://launchpadlibrarian.net/436102255/bwa_0.7.17-1ubuntu0.1_source.changes)

## Binary packages built by this source

[bwa](/ubuntu/bionic/%2Bpackage/bwa):
Burrows-Wheeler Aligner

BWA is a software package for mapping low-divergent sequences against

 a large reference genome, such as the human genome. It consists of

 three algorithms: BWA-backtrack, BWA-SW and BWA-MEM. The first

 algorithm is designed for Illumina sequence reads up to 100bp, while

 the rest two for longer sequences ranged from 70bp to 1Mbp. BWA-MEM

 and BWA-SW share similar features such as long-read support and split

 alignment, but BWA-MEM, which is the latest, is generally recommended

 for high-quality queries as it is faster and more accurate. BWA-MEM

 also has better performance than BWA-backtrack for 70-100bp Illumina

 reads.

[bwa-dbgsym](/ubuntu/bionic/%2Bpackage/bwa-dbgsym):
debug symbols for bwa

[libbwa-dev](/ubuntu/bionic/%2Bpackage/libbwa-dev):
Burrows-Wheeler Aligner source files

BWA is a software package for mapping low-divergent sequences against

 a large reference genome, such as the human genome. It consists of

 three algorithms: BWA-backtrack, BWA-SW and BWA-MEM. The first

 algorithm is designed for Illumina sequence reads up to 100bp, while

 the rest two for longer sequences ranged from 70bp to 1Mbp. BWA-MEM

 and BWA-SW share similar features such as long-read support and split

 alignment, but BWA-MEM, which is the latest, is generally recommended

 for high-quality queries as it is faster and more accurate. BWA-MEM

 also has better performance than BWA-backtrack for 70-100bp Illumina

 reads.

 .

 This package contains the source files for use in other programs.

[![Launchpad](/@@/launchpad-footer-logo.svg)](https://launchpad.net/)
 •
[Take the tour](https://launchpad.net/%2Btour)
 •
[Read the guide](https://help.launchpad.net/)

© 2004
[Canonical Ltd.](http://canonical.com/)
 •
[Terms of use](https://launchpad.net/legal)
 •
[Data privacy](https://www.ubuntu.com/legal/dataprivacy)
 •
[Contact Launchpad Support](/feedback)
 •
[Blog](http://blog.launchpad.net/)
 •
[Careers](https://canonical.com/careers)
 •
[System status](https://ubuntu.social/%40launchpadstatus)
 •
6394e03
([Get the code!](https://dev.launchpad.net/))



=== Content from coreymhudson.github.io_7a238721_20250121_001808.html ===

[Genomic Cybersecurity](/)

# Taking local control of genomics machines through BWA

April 22, 2019

August 12, 2019

## Summary

---

In this post we discuss cybersecurity risks posed to systems engaged in genomic analysis, using a standard **best-practices** genomic pipeline. This post details the existence and consequences of an arbitrary code execution vulnerability affecting the most recent version of BWA on Github, including how it was discovered, exploit development, and demonstration of launching the exploit in a high-fidelity model environment. Because of the standards of storage and processing in pipeline environments, the vulnerability can be exploited to access and modify patient data. The vulnerable code can be reached and exploited by running **“bwa mem”** with a maliciously crafted .ALT file residing locally alongside other legitimate index files. Therefore, the application must be run locally to reach exploitation, but the payload contained in a maliciously crafted .ALT file, may be delivered remotely. The demonstration described in this post of an attack exploiting this vulnerability requires a presence on the target system’s network.

## Implications

---

Because of the risks posed by the vulnerabilities described below, *medical/clinical genomic information*, *intellectual property related to genomic/genetic modification* and *forensic tests based on genomic matching* are potentially at risk if they are processed on servers doing genomic analysis.

## Mitigations

---

In the section below we demonstrate how certain issues of cybersecurity hygiene, in combination with some long-term changes in the standards for hosting data can mitigate these vulnerabilities. In addition, we have submitted an [issue](https://github.com/lh3/bwa/pull/232) and the maintainers have developed a [patch](https://github.com/lh3/bwa/commit/20d0a13092aa4cb73230492b05f9697d5ef0b88e) to the public-facing BWA GitHub page: <https://github.com/lh3/bwa>

This issue has been reported to MITRE/US-CERT/NIST, as [CVE-2019-10269](https://nvd.nist.gov/vuln/detail/CVE-2019-10269).

This report characterizes the vulnerability as critical/high because of its potential to allow unauthorized disclosure of information, unauthorized modification and disruption of service.
## Background on Next Gen Sequencing

---

The low cost of high throughput sequencing, primary by assembly of short genomic reads (known as Next Generation Sequencing or NGS), has fundamentally changed the scale and scope of genomic analysis.[1](#ref1) The main idea behind NGS is to take gigabit scale genomic sequences and cut them into short strings (roughly 100-300 letters long) at high redundancy, with relatively low error rates and then use software to assemble the short strings (known as reads). While the fine-tuning of this process has dramatically reduced the start-to-finish cost of genomics, it has also induced heavy computational burdens. An average human genome may cost only a few hundred dollars to sequence (circa 2018) and assemble in software, but produces several terabytes of data in the process and requires machines with large amount of RAM to complete analysis. The rapid growth of this field has led to a surge of genomic tools and pipelines, many of which do nearly identical tasks and have extremely variable coding standards.[2](#ref2)

![NGS](assets/Whole_genome_shotgun_sequencing_versus_Hierarchical_shotgun_sequencing.png)*Visualization of whole genome sequencing, remixed with permission from Commins, J., Toft, C., Fares, M. A. “Computational Biology Methods and Their Application to the Comparative Genomics of Endocellular Symbiotic Bacteria of Insects.” Biol. Procedures Online (2009). Accessed via Springer Images. Original [here](https://commons.wikimedia.org/wiki/File%3AWhole_genome_shotgun_sequencing_versus_Hierarchical_shotgun_sequencing.png).*

## Best Practices Pipeline for Variant Calling

---

One of the most important uses of whole genome sequencing is variant-calling. Variants can be used to directly observe genetic mutations. This can have clinical, scientific or forensic applications. In variant calling, the data typically arrive to the pipeline as raw reads from an NGS machine and leave the pipeline as a Variant Call Format (VCF) file, which lists each variable position relative to a reference genome.[3](#ref3) The process of associating raw reads with a known reference genome, is referred to as *mapping*.

Different flavors of variant discovery require different steps, depending on if they pool samples, are germline (meaning fixed at birth) or somatic (meaning changing in the body) and whether they are single mutations or variations in copy number. In general, however there is a shared preprocessing set of steps. The best-practices listed below were developed by the BROAD Institute which sequences and assemble large numbers of genomes and has produced some of the most commonly used softwares for variant discovery.[4](#ref4)

![preprocess](assets/dznasg7toiq1.png) *Original protocol* [here](https://software.broadinstitute.org/gatk/best-practices/workflow?id=11165).

Data comes into the preprocessing stages as [FASTQ Files](http://support.illumina.com/content/dam/illumina-support/help/BaseSpaceHelp_v2/Content/Vault/Informatics/Sequencing_Analysis/BS/swSEQ_mBS_FASTQFiles.htm). These files are then processed using a tool called [BWA](https://github.com/lh3/bwa) that performs the task of **“Map to Reference”**. This produces a [BAM File](https://support.illumina.com/help/BS_App_RNASeq_Alignment_OLH_1000000006112/Content/Source/Informatics/BAM-Format.htm). Following this, duplicates are marked and a set of quality scores are estimated using a tool known as [Picard](https://github.com/broadinstitute/picard). The outcome of this is an *Analysis-Ready* BAM file.

![somatic](assets/tzm69d8e2spl.png) This protocol describes the best practices for determining short germline variants - using **BWA**, followed by Picard, then followed by a number of functions within the GATK software. *Original posted* [here.](https://software.broadinstitute.org/gatk/best-practices/workflow?id=11145)

## The vulnerability and exploit

---

Given the above listed best-practices, this blog attempts to determine if it is possible to use an early step in a common software pipeline to exploit the outcome of a genomic test. Furthermore, we try to determine if we can change the raw data, limiting the ability of the user to detect this exploit after the fact. We do not fool ourselves into believing that the exploit we are attempting to craft is novel or particularly complex. The vulnerabilities that are included in this study are well-known, namely use of ARP poisoning to intercept and modify FTP traffic and `getc buffer overflow` vulnerabilities. In simulating our environment, we accept some security assumptions (namely, disabling of ASLR and stack canaries to make the exploit easier to visualize). Other assumptions are made from our experience of how these genomics pipelines are executed in-practice. Assuming .ALT files, and *in fact* many genomic databases, will be delivered over FTP is not only supported as a by-product of BWA’s associated bwa.kit software, but by the standard for holding and delivering genomic databases. The same is true for storing files as read/write-enabled and a lack of file integrity protections such as externally-stored file hashes.

### Background

---

In 2014, the developers of BWA added the ALT-aware capacity.[5](#ref5) This capability allowed for a more statistically robust model for mapping genomic reads to a reference genome. Instead of assuming a canonical genome as a reference, the ALT-aware capability allows users to map reads to a population, which is effectively a distributional representation of the genome. This capacity created the need for a new file in BWA, the .ALT file. This file is constructed using a large collection of genomic variants (a challenge for individual users). And since the target of this file is constantly changing (as new individuals are added to the population of sequenced genomes), there is an advantage to storing these files at a central repository where anyone can access them. For the Human Genome, the US National Center for Biotechnology Information (NCBI) has provided resources for the storage and delivery of these files as a tarred and gzipped directory of indices ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA\_000001405.15\_GRCh38/seqs\_for\_alignment\_pipelines.ucsc\_ids/. BWA provides a tool called bwa.kit which uses this repository to collect the most recent version of the file. BWA requires the program to access a set of index files, one of which, the .ALT file, stores the variable reference genome. When the user runs BWA in mapping mode (BWA MEM), these index files are read into memory first, then the read files are individually mapped to the reference genome. The result is a BAM file that stores the mapping of each read to the reference genome.

### Steps in discovery

---

We discovered an exploitable arbitrary code execution vulnerability in the most recent version of BWA on GitHub while researching the security of BWA’s most common usage, BWA-MEM, in ALT-aware mode.

The vulnerable function allocates a 1024 byte buffer to store the contents of a .ALT file. Because the file size is not checked prior to allocating the buffer, a classic buffer overflow vulnerability exists. BWA must be run with the **mem** command line option to reach the vulnerable code. Additionally, .ann, .amb, .pac, .bwt, .sa, and .ALT files must be present in order for execution to reach the vulnerable code. These index files can be retrieved from running **bwa index** or delivered to the user by NCBI. The developers have acknowledged the **bwa mem** option to be the most commonly used BWA option by researchers. Though this vulnerability was only tested in BWA compiled for 32-bit and 64-bit x86 Linux, it is believed to exist in BWA compiled for any operating system. As indicated in the code comment in the figure below, BWA developers appear to be aware that a bug exists but may not be aware of its repercussions.

![bwaVuln](assets/bwaVuln.png) *Function containing vulnerable code in BWA.*

### Crafting an exploit

---

We make a few crucial security assumptions in crafting this exploit. To craft this exploit we required that ASLR and stack canaries are turned off. While this is not standard practice, it allows us to dramatically ease crafting the exploit for teaching purposes. The second assumption is that FASTQ files are read/write enabled. This *is* typical of normal environments, based on our own experience with how these files are stored on genomic servers.

As mentioned above, the bug can be encountered by using the **bwa mem** command in the same directory as .ann, .amb, .pac, .bwt, .sa files and a .ALT file with a line exceeding 1024 bytes in length. **bwa mem** uses several intermediate index files, the additional files that are required to be present, to allow mapping to the reference genome. BWA takes in the prefix of the index files as well as two FASTQ files. We are running this analysis in paired-end mode, which is the current standard in genomic analysis.

After parsing the command line options to determine its branch of execution, BWA opens and reads each of the index files. After reading the other index files, it attempts to open the .ALT file and store its contents into a locally created, static buffer. The offending function call responsible for writing the file contents into memory is `fgetc()`. The function compares each character as its read in for `EOF, \t, \ n,` or `\r`, in which case it stops writing the current line’s contents to memory. If none of these characters are found it continues reading. Subsequently, providing a file named with the same prefix as the index files but with the .ALT extension that also contains a line made up of more than the allocated 1024 bytes and none of the aforementioned ASCII characters will overflow the buffer.

A user can create a segmentation fault if **bwa mem** is used with the index files and a .ALT file containing a line of 5200 bytes or more. This is shown in the figure below.

![bwaSegFault](assets/bwaSegFault.png)  *Segmentation fault during execution due to size of file. Note: The file consists of only one line with 5200 characters.*

The extra bytes are allocated to store other local variables including some of the index files’ names, a hash map, the incrementor variable indicating the current iteration’s buffer offset, and the .ALT file pointer. The latter three variables are important to note because they will be crucial to the success of the exploit. Running **bwa mem** with the large .ALT file in a debugger (we used GDB) reveals that the program crashes prior to returning from the vulnerable function, actually crashing during a call to `fgetc()`. Looking a little closer, GDB shows that the segmentation fault occurs due to an attempt to read from an invalid memory address (see the figure below).

![firstSegFault64bit](assets/firstSegFault64bit.png) *Segmentation fault during execution due to reading from unmapped (invalid) memory address 0x555555767f stored in RDX+0x10.*

This could be due to the file contents overwriting a local variable specifying where to read, such as the file pointer. Examining memory contents of local variables reveals that the last character written has indeed overwritten the last byte of the file pointer.

![filePointerChange](assets/filePointerChange.png) *First ‘A’ (0x41) byte being written over file pointer value.*

This pointer to the stream must be kept intact to continue reading the file contents into memory. Subsequently, the offset of this local variable from the start of the buffer must be calculated to determine where to write the pointer value. Observing the value of RAX at the time the offending character is written gives the value of the offset stored in the local variable *i*. In this case, writing the address value 0x1428 (5160) bytes into the payload will preserve the pointer. Once the value of the pointer to the file object has been written in the appropriate place, the program continues past this point, but segmentation faults again without control of the instruction pointer.

Looking more closely at the other local variables in memory surrounding the file pointer reveals that the pointer to the hash map and the incrementor variable mentioned above are also being overwritten. As a result, first, any dereference of the hash map pointer will access the wrong memory address (in this case an invalid memory address) and due to the corrupt incrementor value, the file contents are written to an arbitrary offset from the start of the buffer due to the corrupt incrementor value. Because the code checks for a null pointer before cleaning up the hash map pointer, its value can either be replaced with null bytes or preserved as we did with the file pointer. However, the incrementor variable must be set to a valid offset to continue storing the file’s contents and overwrite the return address. If these addresses are written correctly, the exploit can overwrite the return address of the vulnerable function and gain control of execution.

By overflowing the buffer and writing specific values for three local variables, control of the return pointer is achieved, and *arbitrary code can be executed at the privilege level that **bwa mem** was run*. The demo proof of concept exploit created uses a `ret2sys` style payload, passing a `sed` script as the argument to `system()` to perform a search and replace on a string sequence in all .FQ files in a directory. The creation and display of this payload can be found in the figure below. This payload was designed to highlight the issue that storage and processing of data, such as FASTQ files, frequently occur on the same system. As a result, any amount of genomic data on this system could be corrupted. However, the payload could just as easily have launched a shell or performed any operation allowed by the privilege level of the BWA process.

![sedPayload](assets/sedPayload.png) *Payload that uses sed to search and replace a 17-character string in all .FQ files in the directory.*

All in all, the vulnerability in BWA created an opportunity for arbitrary code execution, but the development of the exploit was made significantly simpler by relaxing several modern security protections. Writing a successful exploit for this vulnerability would be prohibitively difficult without local access to the system where BWA was running. First, it would require trial and error to determine the location and values of the local variables to the vulnerable function.

Next, without ASLR being disabled, a Return-Oriented Programming (ROP) chain or some other method to nullify ASLR would have been necessary. If stack smashing protection had been in place, the exploit would have needed to repair the stack canary. Altogether, it is possible to write this exploit and launch it remotely, but it would have been significantly more difficult given a system and compiled application with these protections.

## Creating the patch

The primary recommendation to developers on fixing the bug is simply to check the size of the .ALT file before reading its contents into memory. Allocating a buffer large enough to hold the contents of the file will ensure that the buffer is not overrun and prevent arbitrary code execution. Additionally, it is not standard security practice to use `fgetc()`, `strcat()`, or `strcpy()` because they do not verify the size of the destination buffer prior to execution. If these functions must be used, the size of the destination buffer should be checked before executing each function. If these changes are implemented, this BWA vulnerability should be mitigated. A proposed fix for the vulnerable code can be found in the figure below and has been distributed to the code maintainers.

![bwaProposedCodeFix](assets/bwaProposedCodeFix.png) *Proposed patch.*

## Realistic genomic environment

---

The second major security assumption made in this assessment was that the index files were precompiled and delivered over insecure FTP. To evaluate the implications of this assumption, we created a model environment and attack to demonstrate a real-world scenario wherein the vulnerability could be exploited to modify patient data. This was created using our internal EmulyticsTM capacity for simulation and assessment.[6](#ref6) The model environment consisted of a network of three nodes resembling the client-server relationship that research facilities participate in to acquire reference genomes plus a malicious actor with presence on the research network. In the demonstration a pipeline node equipped with BWA software requests reference genome files from a FTP server in the cloud, and another node existing on the pipeline node’s network is controlled by an attacker and launches an attack with the intent of corrupting patient data.

![FTP_MITM_demo](assets/FTP_MITM_demo.png)  *MitM network demo. GRCh38 is the package of index files delivered by FTP. Blue folder is expected gzipped tarfile and red folder is the maliciously crafted gzipped tarfile.*

We designed an attack scenario around this realistic network interaction involving a Man-in-the-Middle (MitM) of the FTP file transfer. The attacker utilizes Bettercap, an extensible MitM framework written in Ruby, to intercept the files being transferred from the cloud to the pipeline and inject the malicious .ALT payload described in the Exploit Development section. Because of the nuances of FTP, specifically in its usage of two separate connections for file transfer, we had to implement two Bettercap TCP proxy modules served from two MitM sessions. The first session, control connection, needed to change the file size communicated to the client to reflect the injected data and prevent the connection from ending before the entire file was transferred. Fortunately, because the injected data was smaller than the server’s original data in the demo, we did not encounter any TCP FIN packets from the server to modify or drop. The second session, data connection, needed to simply replace the data being sent from the server with our malicious payload.

At the conclusion of the FTP session, the payload is delivered and ready to be launched. Becuase of the legitimate server’s usage of FTP and our modifying traffic in transit, the victim is unaware of the introduction of the malicious payload and unknowingly launches the exploit when BWA-MEM is next run.

![genomics_process](assets/genomics_process.png) *Head to Head description of full genomic pipeline and the impact of BWA indices with incorporated malware. BWA MEM does a search and replace of the sample files to corrupt the results. This results in changes that are propogated through the downstream programs.*

## Reasons for exploitability

---

1. BWA does not fail on poisoned .ALT file
2. Easiest way to collect .ALT files is bwa.kit over FTP
3. Files were not stored as read-only or in read-only directories
4. Files did not have a database to protect them from change
5. .ALT files were not checked after delivery

## Mitigation Recommendations

---

First, our recommendation to mitigate the attack we demonstrated on BWA-MEM is to simply fix the vulnerability. We exploited a classic buffer overflow vulnerability that can easily be resolved using more secure buffer allocation practices as described above and performed elsewhere in BWA’s code. There is currently a patch existing for this bug, and we recommend that users download and install the most recent version of BWA [here](https://github.com/lh3/bwa/tree/master).

Our primary recommendation to mitigate attacks targeting other genomics software or other areas of the pipeline is to separate the storage of data with its processing. Because these are two independent functions, they do not need to occur on the same system. If the patient data (stored as FASTQ files) were not stored on the same system and not write-enabled, the exploit payload would not have been able to modify this data. We believe this separation of responsibilities should be a standard best practice implemented by all facilities performing genomics research and processing.

Another best practice recommendation is to utilize file integrity checks, especially when downloading files over insecure protocols such as FTP or HTTP. Integrity checks such as comparing file hashes using a modern cryptographic hashing algorithm (i.e. SHA-3) will reveal any modification to a .ALT or any other downloaded file as a result of a MitM.

MODIFICATION: As of 3/12/19 NCBI issued MD5 checksums on all of its files stored here. We encourage all users to use them and all softwares downloading data directly from ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA\_000001405.15\_GRCh38/seqs\_for\_alignment\_pipelines.ucsc\_ids/ to check the sums at ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA\_000001405.15\_GRCh38/seqs\_for\_alignment\_pipelines.ucsc\_ids/md5checksums.txt .

## Summary

---

This study demonstrates a bug in the standard genomics pipeline. This bug was easily patched but points to a number of problems in standard genomics proceedure. We believe that fixing these issues will be valuable in making genomics, especially as it concerns patient medical genomics, safer and more secure.

## References

---

* [1] [Stephens, Zachary D., et al. “Big data: astronomical or genomical?.” PLoS biology 13.7 (2015): e1002195](http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1002195)
* [2] [Henry, Vincent J., et al. “OMICtools: an informative directory for multi-omic data analysis.” Database 2014 (2014)](https://academic.oup.com/database/article/doi/10.1093/database/bau069/2634692/OMICtools-an-informative-directory-for-multi-omic)
* [3] [VCF (Variant Call Format) version 4.0](http://www.internationalgenome.org/wiki/Analysis/vcf4.0/)
* [4] [Cornish, Adam, and Chittibabu Guda. “A comparison of variant calling pipelines using genome in a bottle as a reference.” BioMed research international 2015 (2015).](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4619817/pdf/BMRI2015-456479.pdf)
* [5] <https://github.com/lh3/bwa/blob/master/README-alt.md>
* [6] [Urias, V., et al. “Emulytics at Sandia National Laboratories.” MODSIM World (2015)](http://www.modsimworld.org/papers/2015/Emulytics_at_Sandia_National_Laboratories.pdf)

---

## Genomic Cybersecurity

* Genomic Cybersecurity
* cmhudso@sandia.gov

* [coreymhudson](https://github.com/coreymhudson)
* [coreymhudson](https://www.twitter.com/coreymhudson)

This site describes cybersecurity research on genomic systems and platforms.



=== Content from github.com_04670bdb_20250121_022616.html ===

[Skip to content](#start-of-content)

## Navigation Menu

Toggle navigation

[Sign in](/login?return_to=https%3A%2F%2Fgithub.com%2Flh3%2Fbwa%2Fcommit%2F20d0a13092aa4cb73230492b05f9697d5ef0b88e)

* Product

  + [GitHub Copilot
    Write better code with AI](https://github.com/features/copilot)
  + [Security
    Find and fix vulnerabilities](https://github.com/features/security)
  + [Actions
    Automate any workflow](https://github.com/features/actions)
  + [Codespaces
    Instant dev environments](https://github.com/features/codespaces)
  + [Issues
    Plan and track work](https://github.com/features/issues)
  + [Code Review
    Manage code changes](https://github.com/features/code-review)
  + [Discussions
    Collaborate outside of code](https://github.com/features/discussions)
  + [Code Search
    Find more, search less](https://github.com/features/code-search)

  Explore
  + [All features](https://github.com/features)
  + [Documentation](https://docs.github.com)
  + [GitHub Skills](https://skills.github.com)
  + [Blog](https://github.blog)
* Solutions

  By company size
  + [Enterprises](https://github.com/enterprise)
  + [Small and medium teams](https://github.com/team)
  + [Startups](https://github.com/enterprise/startups)
  + [Nonprofits](/solutions/industry/nonprofits)
  By use case
  + [DevSecOps](/solutions/use-case/devsecops)
  + [DevOps](/solutions/use-case/devops)
  + [CI/CD](/solutions/use-case/ci-cd)
  + [View all use cases](/solutions/use-case)

  By industry
  + [Healthcare](/solutions/industry/healthcare)
  + [Financial services](/solutions/industry/financial-services)
  + [Manufacturing](/solutions/industry/manufacturing)
  + [Government](/solutions/industry/government)
  + [View all industries](/solutions/industry)

  [View all solutions](/solutions)
* Resources

  Topics
  + [AI](/resources/articles/ai)
  + [DevOps](/resources/articles/devops)
  + [Security](/resources/articles/security)
  + [Software Development](/resources/articles/software-development)
  + [View all](/resources/articles)

  Explore
  + [Learning Pathways](https://resources.github.com/learn/pathways)
  + [White papers, Ebooks, Webinars](https://resources.github.com)
  + [Customer Stories](https://github.com/customer-stories)
  + [Partners](https://partner.github.com)
  + [Executive Insights](https://github.com/solutions/executive-insights)
* Open Source

  + [GitHub Sponsors
    Fund open source developers](/sponsors)
  + [The ReadME Project
    GitHub community articles](https://github.com/readme)
  Repositories
  + [Topics](https://github.com/topics)
  + [Trending](https://github.com/trending)
  + [Collections](https://github.com/collections)
* Enterprise

  + [Enterprise platform
    AI-powered developer platform](/enterprise)
  Available add-ons
  + [Advanced Security
    Enterprise-grade security features](https://github.com/enterprise/advanced-security)
  + [GitHub Copilot
    Enterprise-grade AI features](/features/copilot#enterprise)
  + [Premium Support
    Enterprise-grade 24/7 support](/premium-support)
* [Pricing](https://github.com/pricing)

Search or jump to...

# Search code, repositories, users, issues, pull requests...

Search

Clear

[Search syntax tips](https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax)

# Provide feedback

We read every piece of feedback, and take your input very seriously.

Include my email address so I can be contacted

  Cancel

 Submit feedback

# Saved searches

## Use saved searches to filter your results more quickly

Name

Query

To see all available qualifiers, see our [documentation](https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax).

  Cancel

 Create saved search

[Sign in](/login?return_to=https%3A%2F%2Fgithub.com%2Flh3%2Fbwa%2Fcommit%2F20d0a13092aa4cb73230492b05f9697d5ef0b88e)

[Sign up](/signup?ref_cta=Sign+up&ref_loc=header+logged+out&ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E%2Fvoltron%2Fcommit_fragments%2Frepo_layout&source=header-repo&source_repo=lh3%2Fbwa)
Reseting focus

You signed in with another tab or window. Reload to refresh your session.
You signed out in another tab or window. Reload to refresh your session.
You switched accounts on another tab or window. Reload to refresh your session.

Dismiss alert

{{ message }}

[lh3](/lh3)
/
**[bwa](/lh3/bwa)**
Public

* [Notifications](/login?return_to=%2Flh3%2Fbwa) You must be signed in to change notification settings
* [Fork
  554](/login?return_to=%2Flh3%2Fbwa)
* [Star
   1.6k](/login?return_to=%2Flh3%2Fbwa)

* [Code](/lh3/bwa)
* [Issues
  196](/lh3/bwa/issues)
* [Pull requests
  44](/lh3/bwa/pulls)
* [Actions](/lh3/bwa/actions)
* [Security](/lh3/bwa/security)
* [Insights](/lh3/bwa/pulse)

Additional navigation options

* [Code](/lh3/bwa)
* [Issues](/lh3/bwa/issues)
* [Pull requests](/lh3/bwa/pulls)
* [Actions](/lh3/bwa/actions)
* [Security](/lh3/bwa/security)
* [Insights](/lh3/bwa/pulse)

## Commit

[Permalink](/lh3/bwa/commit/20d0a13092aa4cb73230492b05f9697d5ef0b88e)

This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.

r1198: exit if .alt is malformatted

[Browse files](/lh3/bwa/tree/20d0a13092aa4cb73230492b05f9697d5ef0b88e)
Browse the repository at this point in the history

```
Resolves [#232](https://github.com/lh3/bwa/pull/232)
```

* Loading branch information

[![@lh3](https://avatars.githubusercontent.com/u/480346?s=40&v=4)](/lh3)

[lh3](/lh3/bwa/commits?author=lh3 "View all commits by lh3")
committed
Jan 23, 2019

1 parent
[f090f93](/lh3/bwa/commit/f090f93460a0379d78a97bfa6133bae5bc8283a6)

commit 20d0a13

 Show file tree

 Hide file tree

Showing
**2 changed files**
with
**8 additions**
and
**2 deletions**.

* Whitespace
* Ignore whitespace

* Split
* Unified

* bntseq.c
  [bntseq.c](#diff-fdf1859e89f3afd4cf70877ac7904f6252df63b0c4177e3119a19491678f4973)
* main.c
  [main.c](#diff-a0cb465674c1b01a07d361f25a0ef2b0214b7dfe9412b7777f89add956da10ec)

## There are no files selected for viewing

8 changes: 7 additions & 1 deletion

8
[bntseq.c](#diff-fdf1859e89f3afd4cf70877ac7904f6252df63b0c4177e3119a19491678f4973 "bntseq.c")

Show comments

[View file](/lh3/bwa/blob/20d0a13092aa4cb73230492b05f9697d5ef0b88e/bntseq.c)
Edit file

Delete file

This file contains bidirectional Unicode text that may be interpreted or compiled differently than what appears below. To review, open the file in an editor that reveals hidden Unicode characters.
[Learn more about bidirectional Unicode characters](https://github.co/hiddenchars)

  [Show hidden characters](%7B%7B%20revealButtonHref%20%7D%7D)

| Original file line number | Diff line number | Diff line change |
| --- | --- | --- |
| Expand Up | | @@ -197,7 +197,13 @@ bntseq\_t \*bns\_restore(const char \*prefix) |
|  |  | } |
|  |  | while (c != '\n' && c != EOF) c = fgetc(fp); |
|  |  | i = 0; |
|  |  | } else str[i++] = c; // FIXME: potential segfault here |
|  |  | } else { |
|  |  | if (i >= 1022) { |
|  |  | fprintf(stderr, "[E::%s] sequence name longer than 1023 characters. Abort!\n", \_\_func\_\_); |
|  |  | exit(1); |
|  |  | } |
|  |  | str[i++] = c; |
|  |  | } |
|  |  | } |
|  |  | kh\_destroy(str, h); |
|  |  | fclose(fp); |
| Expand Down | |  |

2 changes: 1 addition & 1 deletion

2
[main.c](#diff-a0cb465674c1b01a07d361f25a0ef2b0214b7dfe9412b7777f89add956da10ec "main.c")

Show comments

[View file](/lh3/bwa/blob/20d0a13092aa4cb73230492b05f9697d5ef0b88e/main.c)
Edit file

Delete file

This file contains bidirectional Unicode text that may be interpreted or compiled differently than what appears below. To review, open the file in an editor that reveals hidden Unicode characters.
[Learn more about bidirectional Unicode characters](https://github.co/hiddenchars)

  [Show hidden characters](%7B%7B%20revealButtonHref%20%7D%7D)

| Original file line number | Diff line number | Diff line change |
| --- | --- | --- |
| Expand Up | | @@ -4,7 +4,7 @@ |
|  |  | #include "utils.h" |
|  |  |  |
|  |  | #ifndef PACKAGE\_VERSION |
|  |  | #define PACKAGE\_VERSION "0.7.17-r1194-dirty" |
|  |  | #define PACKAGE\_VERSION "0.7.17-r1198-dirty" |
|  |  | #endif |
|  |  |  |
|  |  | int bwa\_fa2pac(int argc, char \*argv[]); |
| Expand Down | |  |

Toggle all file notes
Toggle all file annotations

### 0 comments on commit `20d0a13`

Please
[sign in](/login?return_to=https%3A%2F%2Fgithub.com%2Flh3%2Fbwa%2Fcommit%2F20d0a13092aa4cb73230492b05f9697d5ef0b88e) to comment.

## Footer

© 2025 GitHub, Inc.

### Footer navigation

* [Terms](https://docs.github.com/site-policy/github-terms/github-terms-of-service)
* [Privacy](https://docs.github.com/site-policy/privacy-policies/github-privacy-statement)
* [Security](https://github.com/security)
* [Status](https://www.githubstatus.com/)
* [Docs](https://docs.github.com/)
* [Contact](https://support.github.com?tags=dotcom-footer)
* Manage cookies
* Do not share my personal information

You can’t perform that action at this time.



=== Content from github.com_05a9aa8c_20250121_001809.html ===

[Skip to content](#start-of-content)

## Navigation Menu

Toggle navigation

[Sign in](/login?return_to=https%3A%2F%2Fgithub.com%2Flh3%2Fbwa%2Fpull%2F232)

* Product

  + [GitHub Copilot
    Write better code with AI](https://github.com/features/copilot)
  + [Security
    Find and fix vulnerabilities](https://github.com/features/security)
  + [Actions
    Automate any workflow](https://github.com/features/actions)
  + [Codespaces
    Instant dev environments](https://github.com/features/codespaces)
  + [Issues
    Plan and track work](https://github.com/features/issues)
  + [Code Review
    Manage code changes](https://github.com/features/code-review)
  + [Discussions
    Collaborate outside of code](https://github.com/features/discussions)
  + [Code Search
    Find more, search less](https://github.com/features/code-search)

  Explore
  + [All features](https://github.com/features)
  + [Documentation](https://docs.github.com)
  + [GitHub Skills](https://skills.github.com)
  + [Blog](https://github.blog)
* Solutions

  By company size
  + [Enterprises](https://github.com/enterprise)
  + [Small and medium teams](https://github.com/team)
  + [Startups](https://github.com/enterprise/startups)
  + [Nonprofits](/solutions/industry/nonprofits)
  By use case
  + [DevSecOps](/solutions/use-case/devsecops)
  + [DevOps](/solutions/use-case/devops)
  + [CI/CD](/solutions/use-case/ci-cd)
  + [View all use cases](/solutions/use-case)

  By industry
  + [Healthcare](/solutions/industry/healthcare)
  + [Financial services](/solutions/industry/financial-services)
  + [Manufacturing](/solutions/industry/manufacturing)
  + [Government](/solutions/industry/government)
  + [View all industries](/solutions/industry)

  [View all solutions](/solutions)
* Resources

  Topics
  + [AI](/resources/articles/ai)
  + [DevOps](/resources/articles/devops)
  + [Security](/resources/articles/security)
  + [Software Development](/resources/articles/software-development)
  + [View all](/resources/articles)

  Explore
  + [Learning Pathways](https://resources.github.com/learn/pathways)
  + [White papers, Ebooks, Webinars](https://resources.github.com)
  + [Customer Stories](https://github.com/customer-stories)
  + [Partners](https://partner.github.com)
  + [Executive Insights](https://github.com/solutions/executive-insights)
* Open Source

  + [GitHub Sponsors
    Fund open source developers](/sponsors)
  + [The ReadME Project
    GitHub community articles](https://github.com/readme)
  Repositories
  + [Topics](https://github.com/topics)
  + [Trending](https://github.com/trending)
  + [Collections](https://github.com/collections)
* Enterprise

  + [Enterprise platform
    AI-powered developer platform](/enterprise)
  Available add-ons
  + [Advanced Security
    Enterprise-grade security features](https://github.com/enterprise/advanced-security)
  + [GitHub Copilot
    Enterprise-grade AI features](/features/copilot#enterprise)
  + [Premium Support
    Enterprise-grade 24/7 support](/premium-support)
* [Pricing](https://github.com/pricing)

Search or jump to...

# Search code, repositories, users, issues, pull requests...

Search

Clear

[Search syntax tips](https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax)

# Provide feedback

We read every piece of feedback, and take your input very seriously.

Include my email address so I can be contacted

  Cancel

 Submit feedback

# Saved searches

## Use saved searches to filter your results more quickly

Name

Query

To see all available qualifiers, see our [documentation](https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax).

  Cancel

 Create saved search

[Sign in](/login?return_to=https%3A%2F%2Fgithub.com%2Flh3%2Fbwa%2Fpull%2F232)

[Sign up](/signup?ref_cta=Sign+up&ref_loc=header+logged+out&ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E%2Fvoltron%2Fpull_requests_fragments%2Fpull_request_layout&source=header-repo&source_repo=lh3%2Fbwa)
Reseting focus

You signed in with another tab or window. Reload to refresh your session.
You signed out in another tab or window. Reload to refresh your session.
You switched accounts on another tab or window. Reload to refresh your session.

Dismiss alert

{{ message }}

[lh3](/lh3)
/
**[bwa](/lh3/bwa)**
Public

* [Notifications](/login?return_to=%2Flh3%2Fbwa) You must be signed in to change notification settings
* [Fork
  554](/login?return_to=%2Flh3%2Fbwa)
* [Star
   1.6k](/login?return_to=%2Flh3%2Fbwa)

* [Code](/lh3/bwa)
* [Issues
  196](/lh3/bwa/issues)
* [Pull requests
  44](/lh3/bwa/pulls)
* [Actions](/lh3/bwa/actions)
* [Security](/lh3/bwa/security)
* [Insights](/lh3/bwa/pulse)

Additional navigation options

* [Code](/lh3/bwa)
* [Issues](/lh3/bwa/issues)
* [Pull requests](/lh3/bwa/pulls)
* [Actions](/lh3/bwa/actions)
* [Security](/lh3/bwa/security)
* [Insights](/lh3/bwa/pulse)

New issue

**Have a question about this project?** Sign up for a free GitHub account to open an issue and contact its maintainers and the community.

 [Sign up for GitHub](/signup?return_to=%2Flh3%2Fbwa%2Fissues%2Fnew%2Fchoose)

By clicking “Sign up for GitHub”, you agree to our [terms of service](https://docs.github.com/terms) and
[privacy statement](https://docs.github.com/privacy). We’ll occasionally send you account related emails.

Already on GitHub?
[Sign in](/login?return_to=%2Flh3%2Fbwa%2Fissues%2Fnew%2Fchoose)
to your account

[Jump to bottom](#issue-comment-box)

# Buffer overflow while storing .alt file using bwa mem #232

 Closed

[zachbthomas](/zachbthomas)
wants to merge
1
commit into
[lh3:master](/lh3/bwa/tree/master "lh3/bwa:master")
from
[zachbthomas:alt-bufferoverflow-patch](/zachbthomas/bwa/tree/alt-bufferoverflow-patch "zachbthomas/bwa:alt-bufferoverflow-patch")

 Closed

# [Buffer overflow while storing .alt file using bwa mem](#top) #232

[zachbthomas](/zachbthomas)
wants to merge
1
commit into
[lh3:master](/lh3/bwa/tree/master "lh3/bwa:master")
from
[zachbthomas:alt-bufferoverflow-patch](/zachbthomas/bwa/tree/alt-bufferoverflow-patch "zachbthomas/bwa:alt-bufferoverflow-patch")

[Conversation
0](/lh3/bwa/pull/232)
[Commits
1](/lh3/bwa/pull/232/commits)
[Checks
0](/lh3/bwa/pull/232/checks)
[Files changed](/lh3/bwa/pull/232/files)

## Conversation

This file contains bidirectional Unicode text that may be interpreted or compiled differently than what appears below. To review, open the file in an editor that reveals hidden Unicode characters.
[Learn more about bidirectional Unicode characters](https://github.co/hiddenchars)

  [Show hidden characters](%7B%7B%20revealButtonHref%20%7D%7D)

[![zachbthomas](https://avatars.githubusercontent.com/u/42255542?s=60&v=4)](/zachbthomas)

Copy link

### @zachbthomas **[zachbthomas](/zachbthomas)** commented [Jan 23, 2019](#issue-402367808)

When using bwa mem and an .alt file containing a line greater than 1024 bytes in length, a buffer overflow results from storing .alt file contents into a 1024 byte static buffer. The `fgetc()` function is used in a loop to read an .alt file byte-by-byte. Each byte is manually stored in the static buffer until a line termination character is read. Exceeding the 1024 byte line size limit will overflow the buffer. This buffer overflow can be exploited to potentially gain arbitrary code execution on the system running bwa.

The proposed commits obtain the size of the .alt file and dynamically allocate adequate memory to store the entire file. The `fseeko()` and `ftello()` functions are used to determine the size of the .alt file. The `calloc()` function is used to dynamically allocate and initialize memory equal to the size of the file. Because enough memory is allocated to store the entire .alt file, the buffer overflow is mitigated in the commit code.

Sorry, something went wrong.

All reactions

[![@zachbthomas](https://avatars.githubusercontent.com/u/42255542?s=40&v=4)](/zachbthomas)

`[Allocate enough dynamic memory to store .alt file prior to reading](/lh3/bwa/pull/232/commits/66b70cf67672bd97f801107a7a366d5d5fdbb63a "Allocate enough dynamic memory to store .alt file prior to reading")`

`[66b70cf](/lh3/bwa/pull/232/commits/66b70cf67672bd97f801107a7a366d5d5fdbb63a)`

[![@lh3](https://avatars.githubusercontent.com/u/480346?s=40&v=4)](/lh3)
[lh3](/lh3)
closed this
in
`[20d0a13](/lh3/bwa/commit/20d0a13092aa4cb73230492b05f9697d5ef0b88e)`
[Jan 23, 2019](#event-2092478460)

This was referenced Jul 4, 2019

[Update bwa to post-january 2019 version
LUMC/hutspot#7](/LUMC/hutspot/issues/7)

Open

[Is any new release planned?
#245](/lh3/bwa/issues/245)

Open

[Sign up for free](/join?source=comment-repo)
**to join this conversation on GitHub**.
Already have an account?
[Sign in to comment](/login?return_to=https%3A%2F%2Fgithub.com%2Flh3%2Fbwa%2Fpull%2F232)

Reviewers

No reviews

Assignees

No one assigned

Labels

None yet

Projects

None yet

Milestone

No milestone

Development

Successfully merging this pull request may close these issues.

1 participant

[![@zachbthomas](https://avatars.githubusercontent.com/u/42255542?s=52&v=4)](/zachbthomas)

Add this suggestion to a batch that can be applied as a single commit.
This suggestion is invalid because no changes were made to the code.
Suggestions cannot be applied while the pull request is closed.
Suggestions cannot be applied while viewing a subset of changes.
Only one suggestion per line can be applied in a batch.
Add this suggestion to a batch that can be applied as a single commit.
Applying suggestions on deleted lines is not supported.
You must change the existing code in this line in order to create a valid suggestion.
Outdated suggestions cannot be applied.
This suggestion has been applied or marked resolved.
Suggestions cannot be applied from pending reviews.
Suggestions cannot be applied on multi-line comments.
Suggestions cannot be applied while the pull request is queued to merge.
Suggestion cannot be applied right now. Please check back later.

## Footer

© 2025 GitHub, Inc.

### Footer navigation

* [Terms](https://docs.github.com/site-policy/github-terms/github-terms-of-service)
* [Privacy](https://docs.github.com/site-policy/privacy-policies/github-privacy-statement)
* [Security](https://github.com/security)
* [Status](https://www.githubstatus.com/)
* [Docs](https://docs.github.com/)
* [Contact](https://support.github.com?tags=dotcom-footer)
* Manage cookies
* Do not share my personal information

You can’t perform that action at this time.



=== Content from github.com_2960cba5_20250121_022617.html ===

[Skip to content](#start-of-content)

## Navigation Menu

Toggle navigation

[Sign in](/login?return_to=https%3A%2F%2Fgithub.com%2Flh3%2Fbwa)

* Product

  + [GitHub Copilot
    Write better code with AI](https://github.com/features/copilot)
  + [Security
    Find and fix vulnerabilities](https://github.com/features/security)
  + [Actions
    Automate any workflow](https://github.com/features/actions)
  + [Codespaces
    Instant dev environments](https://github.com/features/codespaces)
  + [Issues
    Plan and track work](https://github.com/features/issues)
  + [Code Review
    Manage code changes](https://github.com/features/code-review)
  + [Discussions
    Collaborate outside of code](https://github.com/features/discussions)
  + [Code Search
    Find more, search less](https://github.com/features/code-search)

  Explore
  + [All features](https://github.com/features)
  + [Documentation](https://docs.github.com)
  + [GitHub Skills](https://skills.github.com)
  + [Blog](https://github.blog)
* Solutions

  By company size
  + [Enterprises](https://github.com/enterprise)
  + [Small and medium teams](https://github.com/team)
  + [Startups](https://github.com/enterprise/startups)
  + [Nonprofits](/solutions/industry/nonprofits)
  By use case
  + [DevSecOps](/solutions/use-case/devsecops)
  + [DevOps](/solutions/use-case/devops)
  + [CI/CD](/solutions/use-case/ci-cd)
  + [View all use cases](/solutions/use-case)

  By industry
  + [Healthcare](/solutions/industry/healthcare)
  + [Financial services](/solutions/industry/financial-services)
  + [Manufacturing](/solutions/industry/manufacturing)
  + [Government](/solutions/industry/government)
  + [View all industries](/solutions/industry)

  [View all solutions](/solutions)
* Resources

  Topics
  + [AI](/resources/articles/ai)
  + [DevOps](/resources/articles/devops)
  + [Security](/resources/articles/security)
  + [Software Development](/resources/articles/software-development)
  + [View all](/resources/articles)

  Explore
  + [Learning Pathways](https://resources.github.com/learn/pathways)
  + [White papers, Ebooks, Webinars](https://resources.github.com)
  + [Customer Stories](https://github.com/customer-stories)
  + [Partners](https://partner.github.com)
  + [Executive Insights](https://github.com/solutions/executive-insights)
* Open Source

  + [GitHub Sponsors
    Fund open source developers](/sponsors)
  + [The ReadME Project
    GitHub community articles](https://github.com/readme)
  Repositories
  + [Topics](https://github.com/topics)
  + [Trending](https://github.com/trending)
  + [Collections](https://github.com/collections)
* Enterprise

  + [Enterprise platform
    AI-powered developer platform](/enterprise)
  Available add-ons
  + [Advanced Security
    Enterprise-grade security features](https://github.com/enterprise/advanced-security)
  + [GitHub Copilot
    Enterprise-grade AI features](/features/copilot#enterprise)
  + [Premium Support
    Enterprise-grade 24/7 support](/premium-support)
* [Pricing](https://github.com/pricing)

Search or jump to...

# Search code, repositories, users, issues, pull requests...

Search

Clear

[Search syntax tips](https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax)

# Provide feedback

We read every piece of feedback, and take your input very seriously.

Include my email address so I can be contacted

  Cancel

 Submit feedback

# Saved searches

## Use saved searches to filter your results more quickly

Name

Query

To see all available qualifiers, see our [documentation](https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax).

  Cancel

 Create saved search

[Sign in](/login?return_to=https%3A%2F%2Fgithub.com%2Flh3%2Fbwa)

[Sign up](/signup?ref_cta=Sign+up&ref_loc=header+logged+out&ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&source=header-repo&source_repo=lh3%2Fbwa)
Reseting focus

You signed in with another tab or window. Reload to refresh your session.
You signed out in another tab or window. Reload to refresh your session.
You switched accounts on another tab or window. Reload to refresh your session.

Dismiss alert

{{ message }}

[lh3](/lh3)
/
**[bwa](/lh3/bwa)**
Public

* [Notifications](/login?return_to=%2Flh3%2Fbwa) You must be signed in to change notification settings
* [Fork
  554](/login?return_to=%2Flh3%2Fbwa)
* [Star
   1.6k](/login?return_to=%2Flh3%2Fbwa)

Burrow-Wheeler Aligner for short-read alignment (see minimap2 for long-read alignment)

### License

[GPL-3.0 license](/lh3/bwa/blob/master/COPYING)

[1.6k
stars](/lh3/bwa/stargazers) [554
forks](/lh3/bwa/forks) [Branches](/lh3/bwa/branches) [Tags](/lh3/bwa/tags) [Activity](/lh3/bwa/activity)
 [Star](/login?return_to=%2Flh3%2Fbwa)

 [Notifications](/login?return_to=%2Flh3%2Fbwa) You must be signed in to change notification settings

* [Code](/lh3/bwa)
* [Issues
  196](/lh3/bwa/issues)
* [Pull requests
  44](/lh3/bwa/pulls)
* [Actions](/lh3/bwa/actions)
* [Security](/lh3/bwa/security)
* [Insights](/lh3/bwa/pulse)

Additional navigation options

* [Code](/lh3/bwa)
* [Issues](/lh3/bwa/issues)
* [Pull requests](/lh3/bwa/pulls)
* [Actions](/lh3/bwa/actions)
* [Security](/lh3/bwa/security)
* [Insights](/lh3/bwa/pulse)

# lh3/bwa

    master[Branches](/lh3/bwa/branches)[Tags](/lh3/bwa/tags)Go to fileCode
## Folders and files

| Name | | Name | Last commit message | Last commit date |
| --- | --- | --- | --- | --- |
| Latest commit History[968 Commits](/lh3/bwa/commits/master/) | | |
| [.github/workflows](/lh3/bwa/tree/master/.github/workflows "This path skips through empty directories") | | [.github/workflows](/lh3/bwa/tree/master/.github/workflows "This path skips through empty directories") |  |  |
| [bwakit](/lh3/bwa/tree/master/bwakit "bwakit") | | [bwakit](/lh3/bwa/tree/master/bwakit "bwakit") |  |  |
| [.gitignore](/lh3/bwa/blob/master/.gitignore ".gitignore") | | [.gitignore](/lh3/bwa/blob/master/.gitignore ".gitignore") |  |  |
| [COPYING](/lh3/bwa/blob/master/COPYING "COPYING") | | [COPYING](/lh3/bwa/blob/master/COPYING "COPYING") |  |  |
| [ChangeLog](/lh3/bwa/blob/master/ChangeLog "ChangeLog") | | [ChangeLog](/lh3/bwa/blob/master/ChangeLog "ChangeLog") |  |  |
| [Makefile](/lh3/bwa/blob/master/Makefile "Makefile") | | [Makefile](/lh3/bwa/blob/master/Makefile "Makefile") |  |  |
| [NEWS.md](/lh3/bwa/blob/master/NEWS.md "NEWS.md") | | [NEWS.md](/lh3/bwa/blob/master/NEWS.md "NEWS.md") |  |  |
| [QSufSort.c](/lh3/bwa/blob/master/QSufSort.c "QSufSort.c") | | [QSufSort.c](/lh3/bwa/blob/master/QSufSort.c "QSufSort.c") |  |  |
| [QSufSort.h](/lh3/bwa/blob/master/QSufSort.h "QSufSort.h") | | [QSufSort.h](/lh3/bwa/blob/master/QSufSort.h "QSufSort.h") |  |  |
| [README-alt.md](/lh3/bwa/blob/master/README-alt.md "README-alt.md") | | [README-alt.md](/lh3/bwa/blob/master/README-alt.md "README-alt.md") |  |  |
| [README.md](/lh3/bwa/blob/master/README.md "README.md") | | [README.md](/lh3/bwa/blob/master/README.md "README.md") |  |  |
| [bamlite.c](/lh3/bwa/blob/master/bamlite.c "bamlite.c") | | [bamlite.c](/lh3/bwa/blob/master/bamlite.c "bamlite.c") |  |  |
| [bamlite.h](/lh3/bwa/blob/master/bamlite.h "bamlite.h") | | [bamlite.h](/lh3/bwa/blob/master/bamlite.h "bamlite.h") |  |  |
| [bntseq.c](/lh3/bwa/blob/master/bntseq.c "bntseq.c") | | [bntseq.c](/lh3/bwa/blob/master/bntseq.c "bntseq.c") |  |  |
| [bntseq.h](/lh3/bwa/blob/master/bntseq.h "bntseq.h") | | [bntseq.h](/lh3/bwa/blob/master/bntseq.h "bntseq.h") |  |  |
| [bwa.1](/lh3/bwa/blob/master/bwa.1 "bwa.1") | | [bwa.1](/lh3/bwa/blob/master/bwa.1 "bwa.1") |  |  |
| [bwa.c](/lh3/bwa/blob/master/bwa.c "bwa.c") | | [bwa.c](/lh3/bwa/blob/master/bwa.c "bwa.c") |  |  |
| [bwa.h](/lh3/bwa/blob/master/bwa.h "bwa.h") | | [bwa.h](/lh3/bwa/blob/master/bwa.h "bwa.h") |  |  |
| [bwamem.c](/lh3/bwa/blob/master/bwamem.c "bwamem.c") | | [bwamem.c](/lh3/bwa/blob/master/bwamem.c "bwamem.c") |  |  |
| [bwamem.h](/lh3/bwa/blob/master/bwamem.h "bwamem.h") | | [bwamem.h](/lh3/bwa/blob/master/bwamem.h "bwamem.h") |  |  |
| [bwamem\_extra.c](/lh3/bwa/blob/master/bwamem_extra.c "bwamem_extra.c") | | [bwamem\_extra.c](/lh3/bwa/blob/master/bwamem_extra.c "bwamem_extra.c") |  |  |
| [bwamem\_pair.c](/lh3/bwa/blob/master/bwamem_pair.c "bwamem_pair.c") | | [bwamem\_pair.c](/lh3/bwa/blob/master/bwamem_pair.c "bwamem_pair.c") |  |  |
| [bwape.c](/lh3/bwa/blob/master/bwape.c "bwape.c") | | [bwape.c](/lh3/bwa/blob/master/bwape.c "bwape.c") |  |  |
| [bwase.c](/lh3/bwa/blob/master/bwase.c "bwase.c") | | [bwase.c](/lh3/bwa/blob/master/bwase.c "bwase.c") |  |  |
| [bwase.h](/lh3/bwa/blob/master/bwase.h "bwase.h") | | [bwase.h](/lh3/bwa/blob/master/bwase.h "bwase.h") |  |  |
| [bwaseqio.c](/lh3/bwa/blob/master/bwaseqio.c "bwaseqio.c") | | [bwaseqio.c](/lh3/bwa/blob/master/bwaseqio.c "bwaseqio.c") |  |  |
| [bwashm.c](/lh3/bwa/blob/master/bwashm.c "bwashm.c") | | [bwashm.c](/lh3/bwa/blob/master/bwashm.c "bwashm.c") |  |  |
| [bwt.c](/lh3/bwa/blob/master/bwt.c "bwt.c") | | [bwt.c](/lh3/bwa/blob/master/bwt.c "bwt.c") |  |  |
| [bwt.h](/lh3/bwa/blob/master/bwt.h "bwt.h") | | [bwt.h](/lh3/bwa/blob/master/bwt.h "bwt.h") |  |  |
| [bwt\_gen.c](/lh3/bwa/blob/master/bwt_gen.c "bwt_gen.c") | | [bwt\_gen.c](/lh3/bwa/blob/master/bwt_gen.c "bwt_gen.c") |  |  |
| [bwt\_lite.c](/lh3/bwa/blob/master/bwt_lite.c "bwt_lite.c") | | [bwt\_lite.c](/lh3/bwa/blob/master/bwt_lite.c "bwt_lite.c") |  |  |
| [bwt\_lite.h](/lh3/bwa/blob/master/bwt_lite.h "bwt_lite.h") | | [bwt\_lite.h](/lh3/bwa/blob/master/bwt_lite.h "bwt_lite.h") |  |  |
| [bwtaln.c](/lh3/bwa/blob/master/bwtaln.c "bwtaln.c") | | [bwtaln.c](/lh3/bwa/blob/master/bwtaln.c "bwtaln.c") |  |  |
| [bwtaln.h](/lh3/bwa/blob/master/bwtaln.h "bwtaln.h") | | [bwtaln.h](/lh3/bwa/blob/master/bwtaln.h "bwtaln.h") |  |  |
| [bwtgap.c](/lh3/bwa/blob/master/bwtgap.c "bwtgap.c") | | [bwtgap.c](/lh3/bwa/blob/master/bwtgap.c "bwtgap.c") |  |  |
| [bwtgap.h](/lh3/bwa/blob/master/bwtgap.h "bwtgap.h") | | [bwtgap.h](/lh3/bwa/blob/master/bwtgap.h "bwtgap.h") |  |  |
| [bwtindex.c](/lh3/bwa/blob/master/bwtindex.c "bwtindex.c") | | [bwtindex.c](/lh3/bwa/blob/master/bwtindex.c "bwtindex.c") |  |  |
| [bwtsw2.h](/lh3/bwa/blob/master/bwtsw2.h "bwtsw2.h") | | [bwtsw2.h](/lh3/bwa/blob/master/bwtsw2.h "bwtsw2.h") |  |  |
| [bwtsw2\_aux.c](/lh3/bwa/blob/master/bwtsw2_aux.c "bwtsw2_aux.c") | | [bwtsw2\_aux.c](/lh3/bwa/blob/master/bwtsw2_aux.c "bwtsw2_aux.c") |  |  |
| [bwtsw2\_chain.c](/lh3/bwa/blob/master/bwtsw2_chain.c "bwtsw2_chain.c") | | [bwtsw2\_chain.c](/lh3/bwa/blob/master/bwtsw2_chain.c "bwtsw2_chain.c") |  |  |
| [bwtsw2\_core.c](/lh3/bwa/blob/master/bwtsw2_core.c "bwtsw2_core.c") | | [bwtsw2\_core.c](/lh3/bwa/blob/master/bwtsw2_core.c "bwtsw2_core.c") |  |  |
| [bwtsw2\_main.c](/lh3/bwa/blob/master/bwtsw2_main.c "bwtsw2_main.c") | | [bwtsw2\_main.c](/lh3/bwa/blob/master/bwtsw2_main.c "bwtsw2_main.c") |  |  |
| [bwtsw2\_pair.c](/lh3/bwa/blob/master/bwtsw2_pair.c "bwtsw2_pair.c") | | [bwtsw2\_pair.c](/lh3/bwa/blob/master/bwtsw2_pair.c "bwtsw2_pair.c") |  |  |
| [code\_of\_conduct.md](/lh3/bwa/blob/master/code_of_conduct.md "code_of_conduct.md") | | [code\_of\_conduct.md](/lh3/bwa/blob/master/code_of_conduct.md "code_of_conduct.md") |  |  |
| [example.c](/lh3/bwa/blob/master/example.c "example.c") | | [example.c](/lh3/bwa/blob/master/example.c "example.c") |  |  |
| [fastmap.c](/lh3/bwa/blob/master/fastmap.c "fastmap.c") | | [fastmap.c](/lh3/bwa/blob/master/fastmap.c "fastmap.c") |  |  |
| [is.c](/lh3/bwa/blob/master/is.c "is.c") | | [is.c](/lh3/bwa/blob/master/is.c "is.c") |  |  |
| [kbtree.h](/lh3/bwa/blob/master/kbtree.h "kbtree.h") | | [kbtree.h](/lh3/bwa/blob/master/kbtree.h "kbtree.h") |  |  |
| [khash.h](/lh3/bwa/blob/master/khash.h "khash.h") | | [khash.h](/lh3/bwa/blob/master/khash.h "khash.h") |  |  |
| [kopen.c](/lh3/bwa/blob/master/kopen.c "kopen.c") | | [kopen.c](/lh3/bwa/blob/master/kopen.c "kopen.c") |  |  |
| [kseq.h](/lh3/bwa/blob/master/kseq.h "kseq.h") | | [kseq.h](/lh3/bwa/blob/master/kseq.h "kseq.h") |  |  |
| [ksort.h](/lh3/bwa/blob/master/ksort.h "ksort.h") | | [ksort.h](/lh3/bwa/blob/master/ksort.h "ksort.h") |  |  |
| [kstring.c](/lh3/bwa/blob/master/kstring.c "kstring.c") | | [kstring.c](/lh3/bwa/blob/master/kstring.c "kstring.c") |  |  |
| [kstring.h](/lh3/bwa/blob/master/kstring.h "kstring.h") | | [kstring.h](/lh3/bwa/blob/master/kstring.h "kstring.h") |  |  |
| [ksw.c](/lh3/bwa/blob/master/ksw.c "ksw.c") | | [ksw.c](/lh3/bwa/blob/master/ksw.c "ksw.c") |  |  |
| [ksw.h](/lh3/bwa/blob/master/ksw.h "ksw.h") | | [ksw.h](/lh3/bwa/blob/master/ksw.h "ksw.h") |  |  |
| [kthread.c](/lh3/bwa/blob/master/kthread.c "kthread.c") | | [kthread.c](/lh3/bwa/blob/master/kthread.c "kthread.c") |  |  |
| [kvec.h](/lh3/bwa/blob/master/kvec.h "kvec.h") | | [kvec.h](/lh3/bwa/blob/master/kvec.h "kvec.h") |  |  |
| [main.c](/lh3/bwa/blob/master/main.c "main.c") | | [main.c](/lh3/bwa/blob/master/main.c "main.c") |  |  |
| [malloc\_wrap.c](/lh3/bwa/blob/master/malloc_wrap.c "malloc_wrap.c") | | [malloc\_wrap.c](/lh3/bwa/blob/master/malloc_wrap.c "malloc_wrap.c") |  |  |
| [malloc\_wrap.h](/lh3/bwa/blob/master/malloc_wrap.h "malloc_wrap.h") | | [malloc\_wrap.h](/lh3/bwa/blob/master/malloc_wrap.h "malloc_wrap.h") |  |  |
| [maxk.c](/lh3/bwa/blob/master/maxk.c "maxk.c") | | [maxk.c](/lh3/bwa/blob/master/maxk.c "maxk.c") |  |  |
| [neon\_sse.h](/lh3/bwa/blob/master/neon_sse.h "neon_sse.h") | | [neon\_sse.h](/lh3/bwa/blob/master/neon_sse.h "neon_sse.h") |  |  |
| [pemerge.c](/lh3/bwa/blob/master/pemerge.c "pemerge.c") | | [pemerge.c](/lh3/bwa/blob/master/pemerge.c "pemerge.c") |  |  |
| [qualfa2fq.pl](/lh3/bwa/blob/master/qualfa2fq.pl "qualfa2fq.pl") | | [qualfa2fq.pl](/lh3/bwa/blob/master/qualfa2fq.pl "qualfa2fq.pl") |  |  |
| [rle.c](/lh3/bwa/blob/master/rle.c "rle.c") | | [rle.c](/lh3/bwa/blob/master/rle.c "rle.c") |  |  |
| [rle.h](/lh3/bwa/blob/master/rle.h "rle.h") | | [rle.h](/lh3/bwa/blob/master/rle.h "rle.h") |  |  |
| [rope.c](/lh3/bwa/blob/master/rope.c "rope.c") | | [rope.c](/lh3/bwa/blob/master/rope.c "rope.c") |  |  |
| [rope.h](/lh3/bwa/blob/master/rope.h "rope.h") | | [rope.h](/lh3/bwa/blob/master/rope.h "rope.h") |  |  |
| [scalar\_sse.h](/lh3/bwa/blob/master/scalar_sse.h "scalar_sse.h") | | [scalar\_sse.h](/lh3/bwa/blob/master/scalar_sse.h "scalar_sse.h") |  |  |
| [utils.c](/lh3/bwa/blob/master/utils.c "utils.c") | | [utils.c](/lh3/bwa/blob/master/utils.c "utils.c") |  |  |
| [utils.h](/lh3/bwa/blob/master/utils.h "utils.h") | | [utils.h](/lh3/bwa/blob/master/utils.h "utils.h") |  |  |
| [xa2multi.pl](/lh3/bwa/blob/master/xa2multi.pl "xa2multi.pl") | | [xa2multi.pl](/lh3/bwa/blob/master/xa2multi.pl "xa2multi.pl") |  |  |
| View all files | | |

## Repository files navigation

* README
* Code of conduct
* GPL-3.0 license

[![Build Status](https://github.com/lh3/bwa/actions/workflows/ci.yaml/badge.svg)](https://github.com/lh3/bwa/actions)
[![SourceForge Downloads](https://camo.githubusercontent.com/e0b0f206b9e4344677a6ce6c1c68b823fc9d4b01d563001332f4d59004d22e54/68747470733a2f2f696d672e736869656c64732e696f2f736f75726365666f7267652f64742f62696f2d6277612e7376673f6c6162656c3d5346253230646f776e6c6f616473)](https://sourceforge.net/projects/bio-bwa/files/?source=navbar)
[![GitHub Downloads](https://camo.githubusercontent.com/892b7ea3b53a79849b0ccc0ccffb17414ca193e9b78b3432ea41b0c8e63a5e66/68747470733a2f2f696d672e736869656c64732e696f2f6769746875622f646f776e6c6f6164732f6c68332f6277612f746f74616c2e7376673f7374796c653d666c6174266c6162656c3d476974487562253230646f776e6c6f616473)](https://github.com/lh3/bwa/releases)
[![BioConda Install](https://camo.githubusercontent.com/1aceb23c47dc46267e28dc1c6e48c6e76d343d4acb9640610983b8116838d397/68747470733a2f2f696d672e736869656c64732e696f2f636f6e64612f646e2f62696f636f6e64612f6277612e7376673f7374796c653d666c6167266c6162656c3d42696f436f6e6461253230696e7374616c6c)](https://anaconda.org/bioconda/bwa)

**Note: [minimap2](https://github.com/lh3/minimap2) has replaced BWA-MEM for PacBio and Nanopore read
alignment.** It retains all major BWA-MEM features, but is ~50 times as fast,
more versatile, more accurate and produces better base-level alignment.
A beta version of [BWA-MEM2](https://github.com/bwa-mem2/bwa-mem2) has been released for short-read mapping.
BWA-MEM2 is about twice as fast as BWA-MEM and outputs near identical alignments.

## Getting started

```
git clone https://github.com/lh3/bwa.git
cd bwa; make
./bwa index ref.fa
./bwa mem ref.fa read-se.fq.gz | gzip -3 > aln-se.sam.gz
./bwa mem ref.fa read1.fq read2.fq | gzip -3 > aln-pe.sam.gz

```

## Introduction

BWA is a software package for mapping DNA sequences against a large reference
genome, such as the human genome. It consists of three algorithms:
BWA-backtrack, BWA-SW and BWA-MEM. The first algorithm is designed for Illumina
sequence reads up to 100bp, while the rest two for longer sequences ranged from
70bp to a few megabases. BWA-MEM and BWA-SW share similar features such as the
support of long reads and chimeric alignment, but BWA-MEM, which is the latest,
is generally recommended as it is faster and more accurate. BWA-MEM also has
better performance than BWA-backtrack for 70-100bp Illumina reads.

For all the algorithms, BWA first needs to construct the FM-index for the
reference genome (the **index** command). Alignment algorithms are invoked with
different sub-commands: **aln/samse/sampe** for BWA-backtrack,
**bwasw** for BWA-SW and **mem** for the BWA-MEM algorithm.

## Availability

BWA is released under [GPLv3](http://en.wikipedia.org/wiki/GNU_General_Public_License). The latest source code is [freely
available at github](https://github.com/lh3/bwa). Released packages can [be downloaded](http://sourceforge.net/projects/bio-bwa/files/) at
SourceForge. After you acquire the source code, simply use `make` to compile
and copy the single executable `bwa` to the destination you want. The only
dependency required to build BWA is [zlib](http://zlib.net/).

Since 0.7.11, precompiled binary for x86\_64-linux is available in [bwakit](http://sourceforge.net/projects/bio-bwa/files/bwakit/).
In addition to BWA, this self-consistent package also comes with bwa-associated
and 3rd-party tools for proper BAM-to-FASTQ conversion, mapping to ALT contigs,
adapter triming, duplicate marking, HLA typing and associated data files.

## Seeking help

The detailed usage is described in the man page available together with the
source code. You can use `man ./bwa.1` to view the man page in a terminal. The
[HTML version](http://bio-bwa.sourceforge.net/bwa.shtml) of the man page can be found at the [BWA website](http://bio-bwa.sourceforge.net/). If you
have questions about BWA, you may [sign up the mailing list](https://lists.sourceforge.net/lists/listinfo/bio-bwa-help) and then send
the questions to bio-bwa-help@sourceforge.net. You may also ask questions
in forums such as [BioStar](http://biostars.org) and [SEQanswers](http://seqanswers.com/).

## Citing BWA

* Li H. and Durbin R. (2009) Fast and accurate short read alignment with
  Burrows-Wheeler transform. *Bioinformatics*, **25**, 1754-1760. [PMID:
  [19451168](http://www.ncbi.nlm.nih.gov/pubmed/19451168)]. (if you use the BWA-backtrack algorithm)
* Li H. and Durbin R. (2010) Fast and accurate long-read alignment with
  Burrows-Wheeler transform. *Bioinformatics*, **26**, 589-595. [PMID:
  [20080505](http://www.ncbi.nlm.nih.gov/pubmed/20080505)]. (if you use the BWA-SW algorithm)
* Li H. (2013) Aligning sequence reads, clone sequences and assembly contigs
  with BWA-MEM. [arXiv:1303.3997v2](http://arxiv.org/abs/1303.3997) [q-bio.GN]. (if you use the BWA-MEM
  algorithm or the **fastmap** command, or want to cite the whole BWA package)

Please note that the last reference is a preprint hosted at [arXiv.org](http://arxiv.org/). I
do not have plan to submit it to a peer-reviewed journal in the near future.

## Frequently asked questions (FAQs)

1. [What types of data does BWA work with?](#type)
2. [Why does a read appear multiple times in the output SAM?](#multihit)
3. [Does BWA work on reference sequences longer than 4GB in total?](#4gb)
4. [Why can one read in a pair has high mapping quality but the other has zero?](#pe0)
5. [How can a BWA-backtrack alignment stands out of the end of a chromosome?](#endref)
6. [Does BWA work with ALT contigs in the GRCh38 release?](#altctg)
7. [Can I just run BWA-MEM against GRCh38+ALT without post-processing?](#postalt)

#### 1. What types of data does BWA work with?

BWA works with a variety types of DNA sequence data, though the optimal
algorithm and setting may vary. The following list gives the recommended
settings:

* Illumina/454/IonTorrent single-end reads longer than ~70bp or assembly
  contigs up to a few megabases mapped to a closely related reference genome:

  ```
    bwa mem ref.fa reads.fq > aln.sam

  ```
* Illumina single-end reads shorter than ~70bp:

  ```
    bwa aln ref.fa reads.fq > reads.sai; bwa samse ref.fa reads.sai reads.fq > aln-se.sam

  ```
* Illumina/454/IonTorrent paired-end reads longer than ~70bp:

  ```
    bwa mem ref.fa read1.fq read2.fq > aln-pe.sam

  ```
* Illumina paired-end reads shorter than ~70bp:

  ```
    bwa aln ref.fa read1.fq > read1.sai; bwa aln ref.fa read2.fq > read2.sai
    bwa sampe ref.fa read1.sai read2.sai read1.fq read2.fq > aln-pe.sam

  ```
* PacBio subreads or Oxford Nanopore reads to a reference genome:

  ```
    bwa mem -x pacbio ref.fa reads.fq > aln.sam
    bwa mem -x ont2d ref.fa reads.fq > aln.sam

  ```

BWA-MEM is recommended for query sequences longer than ~70bp for a variety of
error rates (or sequence divergence). Generally, BWA-MEM is more tolerant with
errors given longer query sequences as the chance of missing all seeds is small.
As is shown above, with non-default settings, BWA-MEM works with Oxford Nanopore
reads with a sequencing error rate over 20%.

#### 2. Why does a read appear multiple times in the output SAM?

BWA-SW and BWA-MEM perform local alignments. If there is a translocation, a gene
fusion or a long deletion, a read bridging the break point may have two hits,
occupying two lines in the SAM output. With the default setting of BWA-MEM, one
and only one line is primary and is soft clipped; other lines are tagged with
0x800 SAM flag (supplementary alignment) and are hard clipped.

#### 3. Does BWA work on reference sequences longer than 4GB in total?

Yes. Since 0.6.x, all BWA algorithms work with a genome with total length over
4GB. However, individual chromosome should not be longer than 2GB.

#### 4. Why can one read in a pair have a high mapping quality but the other has zero?

This is correct. Mapping quality is assigned for individual read, not for a read
pair. It is possible that one read can be mapped unambiguously, but its mate
falls in a tandem repeat and thus its accurate position cannot be determined.

#### 5. How can a BWA-backtrack alignment stand out of the end of a chromosome?

Internally BWA concatenates all reference sequences into one long sequence. A
read may be mapped to the junction of two adjacent reference sequences. In this
case, BWA-backtrack will flag the read as unmapped (0x4), but you will see
position, CIGAR and all the tags. A similar issue may occur to BWA-SW alignment
as well. BWA-MEM does not have this problem.

#### 6. Does BWA work with ALT contigs in the GRCh38 release?

Yes, since 0.7.11, BWA-MEM officially supports mapping to GRCh38+ALT.
BWA-backtrack and BWA-SW don't properly support ALT mapping as of now. Please
see [README-alt.md](https://github.com/lh3/bwa/blob/master/README-alt.md) for details. Briefly, it is recommended to use
[bwakit](http://sourceforge.net/projects/bio-bwa/files/bwakit/), the binary release of BWA, for generating the reference genome
and for mapping.

#### 7. Can I just run BWA-MEM against GRCh38+ALT without post-processing?

If you are not interested in hits to ALT contigs, it is okay to run BWA-MEM
without post-processing. The alignments produced this way are very close to
alignments against GRCh38 without ALT contigs. Nonetheless, applying
post-processing helps to reduce false mappings caused by reads from the
diverged part of ALT contigs and also enables HLA typing. It is recommended to
run the post-processing script.

## About

Burrow-Wheeler Aligner for short-read alignment (see minimap2 for long-read alignment)

### Topics

[bioinformatics](/topics/bioinformatics "Topic: bioinformatics")
[genomics](/topics/genomics "Topic: genomics")
[sequence-alignment](/topics/sequence-alignment "Topic: sequence-alignment")
[fm-index](/topics/fm-index "Topic: fm-index")

### Resources

[Readme](#readme-ov-file)
### License

[GPL-3.0 license](#GPL-3.0-1-ov-file)
### Code of conduct

[Code of conduct](#coc-ov-file)

[Activity](/lh3/bwa/activity)
### Stars

[**1.6k**
stars](/lh3/bwa/stargazers)
### Watchers

[**107**
watching](/lh3/bwa/watchers)
### Forks

[**554**
forks](/lh3/bwa/forks)
[Report repository](/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Flh3%2Fbwa&report=lh3+%28user%29)

## [Releases 5](/lh3/bwa/releases)

[BWA-0.7.18 (r1243)
Latest

Apr 15, 2024](/lh3/bwa/releases/tag/v0.7.18)
[+ 4 releases](/lh3/bwa/releases)

## [Packages 0](/users/lh3/packages?repo_name=bwa)

## [Contributors 25](/lh3/bwa/graphs/contributors)

* [![@lh3](https://avatars.githubusercontent.com/u/480346?s=64&v=4)](https://github.com/lh3)
* [![@jmarshall](https://avatars.githubusercontent.com/u/70921?s=64&v=4)](https://github.com/jmarshall)
* [![@nh13](https://avatars.githubusercontent.com/u/846274?s=64&v=4)](https://github.com/nh13)
* [![@RoelKluin](https://avatars.githubusercontent.com/u/596671?s=64&v=4)](https://github.com/RoelKluin)
* [![@rmzelle](https://avatars.githubusercontent.com/u/77951?s=64&v=4)](https://github.com/rmzelle)
* [![@jdagilliland](https://avatars.githubusercontent.com/u/1383440?s=64&v=4)](https://github.com/jdagilliland)
* [![@srl147](https://avatars.githubusercontent.com/u/1560612?s=64&v=4)](https://github.com/srl147)
* [![@bwlang](https://avatars.githubusercontent.com/u/61636?s=64&v=4)](https://github.com/bwlang)
* [![@martin-g](https://avatars.githubusercontent.com/u/232002?s=64&v=4)](https://github.com/martin-g)
* [![@sjackman](https://avatars.githubusercontent.com/u/291551?s=64&v=4)](https://github.com/sjackman)
* [![@ramyala](https://avatars.githubusercontent.com/u/602627?s=64&v=4)](https://github.com/ramyala)
* [![@wookietreiber](https://avatars.githubusercontent.com/u/827088?s=64&v=4)](https://github.com/wookietreiber)
* [![@jwbargsten](https://avatars.githubusercontent.com/u/919322?s=64&v=4)](https://github.com/jwbargsten)
* [![@mfcovington](https://avatars.githubusercontent.com/u/1494842?s=64&v=4)](https://github.com/mfcovington)

[+ 11 contributors](/lh3/bwa/graphs/contributors)

## Languages

* [C
  86.4%](/lh3/bwa/search?l=c)
* [JavaScript
  5.6%](/lh3/bwa/search?l=javascript)
* [Roff
  4.2%](/lh3/bwa/search?l=roff)
* [C++
  1.5%](/lh3/bwa/search?l=c%2B%2B)
* [Perl
  1.2%](/lh3/bwa/search?l=perl)
* [Shell
  0.6%](/lh3/bwa/search?l=shell)
* [Makefile
  0.5%](/lh3/bwa/search?l=makefile)

## Footer

© 2025 GitHub, Inc.

### Footer navigation

* [Terms](https://docs.github.com/site-policy/github-terms/github-terms-of-service)
* [Privacy](https://docs.github.com/site-policy/privacy-policies/github-privacy-statement)
* [Security](https://github.com/security)
* [Status](https://www.githubstatus.com/)
* [Docs](https://docs.github.com/)
* [Contact](https://support.github.com?tags=dotcom-footer)
* Manage cookies
* Do not share my personal information

You can’t perform that action at this time.



=== Content from usn.ubuntu.com_57898f9c_20250121_001810.html ===


Your submission was sent successfully!
*Close*

Thank you for contacting us. A member of our team will be in touch shortly.
*Close*

You have successfully unsubscribed!
*Close*

Thank you for signing up for our newsletter!

In these regular emails you will find the latest updates about
Ubuntu and upcoming events where you can meet our team.*Close*

Your preferences have been successfully updated. *Close*

[Canonical Ubuntu](/)

* [Menu](/navigation)

* [Products](/navigation#products-navigation)
* [Use cases](/navigation#use-case-navigation)
* [Support](/navigation#support-navigation)
* [Community](/navigation#community-navigation)
* [Get Ubuntu](/navigation#get-ubuntu-navigation)

[![](https://assets.ubuntu.com/v1/82818827-CoF_white.svg)

Security](/security)

* [ESM](/security/esm)
* [Livepatch](/security/livepatch)
* [Certifications & Hardening](/security/compliance-automation)
* [CVEs](/security/cves)
* [Notices](/security/notices)
* [Docker Images](/security/docker-images)

# USN-4087-1: BWA vulnerability

6 August 2019

BWA could be made to crash or run programs as your login if it
opened a specially crafted file.

### Reduce your security exposure

Ubuntu Pro provides ten-year security coverage to 25,000+ packages in Main and Universe repositories, and it is free for up to five machines.

[Learn more about Ubuntu Pro](/pro)

## Releases

* [Ubuntu 19.04](/security/notices?release=disco)
* [Ubuntu 18.04 ESM](/security/notices?release=bionic)

## Packages

* [bwa](/security/cves?package=bwa) - Software package for mapping DNA sequences against a large reference genome

## Details

It was discovered that Burrows-Wheeler Aligner (BWA) mishandled certain

crafted .alt files. An attacker could use this vulnerability to cause a

denial of service (crash) or possibly execute arbitrary code.

### Reduce your security exposure

Ubuntu Pro provides ten-year security coverage to 25,000+ packages in Main and Universe repositories, and it is free for up to five machines.

[Learn more about Ubuntu Pro](/pro)

## Update instructions

The problem can be corrected by updating your system to the following package versions:

##### Ubuntu 19.04

* [bwa](https://launchpad.net/ubuntu/%2Bsource/bwa)
  -
  [0.7.17-3~ubuntu0.19.04.1](https://launchpad.net/ubuntu/%2Bsource/bwa/0.7.17-3~ubuntu0.19.04.1)

##### Ubuntu 18.04

* [bwa](https://launchpad.net/ubuntu/%2Bsource/bwa)
  -
  [0.7.17-1ubuntu0.1](https://launchpad.net/ubuntu/%2Bsource/bwa/0.7.17-1ubuntu0.1)

In general, a standard system update will make all the necessary changes.

## References

* [CVE-2019-10269](/security/CVE-2019-10269)

## Related notices

* [USN-4857-1](/security/notices/USN-4857-1)

### Join the discussion

* [Ubuntu security updates mailing list](https://lists.ubuntu.com/mailman/listinfo/ubuntu-hardened)
* [Security announcements mailing list](https://lists.ubuntu.com/mailman/listinfo/ubuntu-security-announce)

### Need help with your security needs?

Ubuntu Pro provides up to ten-year security coverage for over 23,000 open-source packages within the Ubuntu Main and Universe repositories.

[Talk to an expert to find out what would work best for you](/contact-us/form?product=pro)

### Further reading

* *Loading...*

---

## [OpenStack](/openstack) [OpenStack](/openstack)

* [What is OpenStack](/openstack/what-is-openstack)
* [Features](/openstack/features)
* [Managed](/openstack/managed)
* [Consulting](/openstack/consulting)
* [Install](/openstack/install)
* [Support](/openstack/support)

---

## [Ceph](/ceph) [Ceph](/ceph)

* [What is Ceph](/ceph/what-is-ceph)
* [Managed](/ceph/managed)
* [Consulting](/ceph/consulting)
* [Docs](/ceph/docs)
* [Install](/ceph/install)

---

## [Kubernetes](/kubernetes) [Kubernetes](/kubernetes)

* [What is Kubernetes](/kubernetes/what-is-kubernetes)
* [Charmed Kubernetes](/kubernetes/charmed-k8s)
* [Managed](/kubernetes/managed)
* [Install](/kubernetes/install)
* [Docs](/kubernetes/docs)
* [Resources](/kubernetes/resources)

---

## [Managed Services](/managed) [Managed Services](/managed)

* [OpenStack](/openstack/managed)
* [Kubernetes](/kubernetes/managed)
* [Ceph](/ceph/managed)
* [Apps](/managed/apps)
* [Observability](/observability/managed)
* [Firefighting](/managed/firefighting-support)

---

## [AI / ML](/ai) [AI / ML](/ai)

* [MLOps](/ai/mlops)
* [Kubeflow](/ai/what-is-kubeflow)
* [MLflow](/ai/mlflow)
* [Consulting](/ai/consulting)
* [Data Science](/ai/data-science)
* [MLOps workshop](/ai/mlops-workshop)

---

## [Robotics](/robotics) [Robotics](/robotics)

* [What is ROS](/robotics/what-is-ros)
* [ROS ESM](/robotics/ros-esm)
* [Community](/robotics/community)
* [Docs](/robotics/docs)

---

## [IoT](/internet-of-things) [IoT](/internet-of-things)

* [App store](/internet-of-things/appstore)
* [Embedded Linux](/embedded)
* [Management](/internet-of-things/management)

---

## [Ubuntu Core](/core) [Ubuntu Core](/core)

* [Features](/core/features)
* [Success stories](/core/stories)
* [Services](/core/services)
* [Docs](/core/docs)

---

## [Ubuntu Desktop](/desktop) [Ubuntu Desktop](/desktop)

* [Organisations](/desktop/organisations)
* [Developers](/desktop/developers)
* [Flavours](/desktop/flavours)
* [WSL](/desktop/wsl)

---

## [Ubuntu Server](/server) [Ubuntu Server](/server)

* [Hyperscale](/server/hyperscale)
* [Docs](/server/docs)

---

## [Cloud](/cloud) [Cloud](/cloud)

* [What is cloud computing](/cloud/cloud-computing)
* [What is private cloud](/cloud/private-cloud)
* [What is hybrid cloud](/cloud/hybrid-cloud)
* [What is multi-cloud](/cloud/multi-cloud)
* [Public cloud](/cloud/public-cloud)

---

## [Security](/security) [Security](/security)

* [ESM](/security/esm)
* [Livepatch](/security/livepatch)
* [Certifications & Hardening](/security/compliance-automation)
* [CVEs](/security/cves)
* [Notices](/security/notices)
* [Docker Images](/security/docker-images)

---

## [Landscape](/landscape) [Landscape](/landscape)

* [Features](/landscape/features)
* [Managed](/landscape/managed)
* [Compare](/landscape/compare)
* [Install](/landscape/docs/quickstart-deployment)
* [Docs](/landscape/docs)
* [Log in to Landscape](https://landscape.canonical.com/)

---

## [Containers](/containers) [Containers](/containers)

* [What are containers](/containers/what-are-containers)
* [Chiseled Ubuntu](/containers/chiseled)
* [Chiseled and .NET](/containers/chiseled/dotnet)

---

## [Downloads](/download) [Downloads](/download)

* [Desktop](/download/desktop)
* [Server](/download/server)
* [Core](/download/core)
* [Cloud](/download/cloud)

---

## [Support](/support) [Support](/support)

* [Your subscriptions](/pro/dashboard)
* [Account users](/pro/users)
* [Pricing](/pricing/pro)
* [Discourse](https://discourse.ubuntu.com/c/project/ubuntu-pro/116/)

---

## [Observability](/observability) [Observability](/observability)

* [What is observability](/observability/what-is-observability)
* [Managed](/observability/managed)

---

## [Pricing](/pricing) [Pricing](/pricing)

* [Consulting](/pricing/consulting)
* [Desktops](/pricing/desktop)
* [Devices](/pricing/devices)

---

## Solutions

* [AI](https://canonical.com/solutions/ai)
* [Data](https://canonical.com/data)
* [Infrastructure](https://canonical.com/solutions/infrastructure)
* [Secure open source](https://canonical.com/solutions/secure-open-source)

---

## Sectors

* [Automotive](/automotive)
* [Industrial](/industrial)
* [Government](/gov)
* [Telco](/telco)
* [Finance](/financial-services)

---

[Contact us](/contact-us)

* [About us](/about)
* [Community](/community)
* [Careers](https://www.canonical.com/careers)
* [Blog](/blog)
* [Resources](/engage)
* [Press centre](/blog/press-centre)

---

© 2025 Canonical Ltd.

Ubuntu and Canonical are registered trademarks of Canonical Ltd.

---

* [Legal information](/legal)
* [Data privacy](/legal/data-privacy)
* Manage your tracker settings
* [Report a bug on this site](https://github.com/canonical/ubuntu.com/issues/new?template=ISSUE_TEMPLATE.yaml)

Back to top

Go to the top of the page



=== Content from lists.ubuntu.com_be8ed54b_20250121_022617.html ===


| **ubuntu-hardened -- Ubuntu security discussion** | |
| --- | --- |
|  | |
| **About ubuntu-hardened** | English (USA)   Chinese (Taiwan) |
| Discussion on Ubuntu security updates, development of new security technologies, and general "hardening" of the operating system.  A place to talk about MAC systems (SELinux, AppArmor, Smack) in Ubuntu, about both kernel-level and user-land security improvements and technologies, etc.  To see the collection of prior postings to the list, visit the [ubuntu-hardened Archives](https://lists.ubuntu.com/archives/ubuntu-hardened). | |
| **Using ubuntu-hardened** | |
| To post a message to all the list members, send email to ubuntu-hardened@lists.ubuntu.com. You can subscribe to the list, or change your existing subscription, in the sections below. | |
| **Subscribing to ubuntu-hardened** | |
| Subscribe to ubuntu-hardened by filling out the following form. You will be sent email requesting confirmation, to prevent others from gratuitously subscribing you. This is a hidden list, which means that the list of members is available only to the list administrator.    | Your email address: |  |  | | --- | --- | --- | | Your name (optional): |  |  | | You may enter a privacy password below. This provides only mild security, but should prevent others from messing with your subscription. **Do not use a valuable password** as it will occasionally be emailed back to you in cleartext. If you choose not to enter a password, one will be automatically generated for you, and it will be sent to you once you've confirmed your subscription. You can always request a mail-back of your password when you edit your personal options. | | | | Pick a password: |  |  | | Reenter password to confirm: |  |  | | Which language do you prefer to display your messages? | English (USA)   Chinese (Taiwan) |  | | Would you like to receive list mail batched in a daily digest? | No  Yes | |  | | | | |

| **ubuntu-hardened Subscribers** | |
| (*The subscribers list is only available to the list administrator.*) Enter your admin address and password to visit the subscribers list: Admin address: Password:  To unsubscribe from ubuntu-hardened, get a password reminder, or change your subscription options enter your subscription email address: If you leave the field blank, you will be prompted for your email address | |

---

[ubuntu-hardened](../listinfo/ubuntu-hardened) list run by jamie at ubuntu.com, mdeslaur at ubuntu.com, sbeattie at ubuntu.com, alex.murray at canonical.com
[ubuntu-hardened administrative interface](../admin/ubuntu-hardened) (requires authorization)
[Overview of all lists.ubuntu.com mailing lists](../listinfo)

| [Delivered by Mailmanversion 2.1.20](http://www.gnu.org/software/mailman/index.html) | [Python Powered](http://www.python.org/) | [GNU's Not Unix](http://www.gnu.org/) | [Debian Powered](http://www.debian.org/) |
| --- | --- | --- | --- |

[![](http://tuxedo-es.org/images/hardened_debian.png)](http://www.debian-hardened.org)


