=== Content from secunia.com_ef67f9c6_20250124_223459.html ===


[Skip to main content](#main-content)
[![Home](/themes/custom/flexera/images/logo.svg)](https://www.flexera.com/)

[![Home](/themes/custom/flexera/images/logo.svg)](https://www.flexera.com/)

Search

## Main navigation

* Solutions
  + Column 1
    - Business challenge
      * [Software renewals and audits](https://www.flexera.com/solutions/software-renewals-audits)
      * [Software license management and optimization](https://www.flexera.com/solutions/software-usage-costs)
      * [SaaS spend management](https://www.flexera.com/solutions/saas-spend)
      * [Cloud cost management](https://www.flexera.com/solutions/cloud-cost)
      * [IT asset lifecycle management](https://www.flexera.com/solutions/it-asset-lifecycle)
      * [CMDB data quality](https://www.flexera.com/solutions/cmdb-data-quality)
      * [Accurate IT inventory](https://www.flexera.com/solutions/it-inventory)
      * [Security and regulatory risk management](https://www.flexera.com/solutions/it-security-regulatory-risk)
      * [Sustainable IT](https://www.flexera.com/solutions/sustainable-it)
      * [AI-powered transformation](https://www.flexera.com/solutions/ai-powered-transformation)
      * [Public sector](https://www.flexera.com/solutions/public-sector)
  + Column 2
    - Spend management by vendor
      * [IBM](https://www.flexera.com/solutions/vendor/ibm)
      * [Oracle](https://www.flexera.com/solutions/vendor/oracle)
      * [Microsoft](https://www.flexera.com/solutions/vendor/microsoft)
      * [SAP](https://www.flexera.com/solutions/vendor/sap)
      * [VMware](https://www.flexera.com/solutions/vendor/vmware)
      * [ServiceNow](https://www.flexera.com/solutions/vendor/servicenow)
      * [AWS](https://www.flexera.com/solutions/vendor/aws)
      * [Salesforce](https://www.flexera.com/solutions/vendor/salesforce)
      * [BMC](https://www.flexera.com/solutions/cmdb-data-quality/bmc)
      * [Adobe](https://www.flexera.com/solutions/vendor/adobe)

  ### Achieve more through a united FinOps and ITAM function

  The future is hybrid. Break down the walls between ITAM and FinOps to drive more revenue, more customer growth and more innovation.

  [Discover More](https://www.flexera.com/resources/hybrid-itam-finops)
* Products
  + Column 1
    - [Flexera One](https://www.flexera.com/products/flexera-one)
      * [IT Visibility](https://www.flexera.com/products/flexera-one/it-visibility)
      * [ITAM](https://www.flexera.com/products/flexera-one/it-asset-management)
      * [SaaS Management](https://www.flexera.com/products/flexera-one/saas-management)
      * [FinOps](https://www.flexera.com/products/flexera-one/finops)
      * [Technology Intelligence Platform](https://www.flexera.com/products/flexera-one/technology-intelligence-platform)
  + Column 2
    - [Snow Atlas](https://www.flexera.com/products/snow-atlas)
      * [Snow Spend Optimizer](https://www.flexera.com/products/snow-atlas/snow-spend-optimizer)
      * [Snow SaaS Management](https://www.flexera.com/products/snow-atlas/snow-saas-management)
  + Column 3
    - Hide group
      * [Security](https://www.flexera.com/products/security)
      * [Application Readiness](https://www.flexera.com/products/adminstudio)
      * [All products](https://www.flexera.com/products)
      * [All Snow products](https://www.flexera.com/products/snow)
      * [Integrations](https://www.flexera.com/products/integrations)

  ### Flexera 2024 State of the Cloud Report

  What do transformative initiatives such as GenAI, machine learning and sustainability mean for the cloud? Check out the 2024 State of the Cloud Report to find the answer as well as all the latest cloud computing trends.

  [View Report](https://info.flexera.com/CM-REPORT-State-of-the-Cloud)
* Success
  + Column 1
    - [Customer success](https://www.flexera.com/customer-success)
      * Support
        + [Flexera support portal](https://community.flexera.com/s/support-hub)
        + [Flexera product documentation](https://docs.flexera.com)
        + [Snow product documentation](https://docs.snowsoftware.io/)
      * Services and training
        + [Services](https://www.flexera.com/customer-success/services)
        + [Training](https://www.flexera.com/customer-success/training)
  + Column 2
    - Hide group
      * [Technology Intelligence Awards](https://www.flexera.com/customer-success/awards)
      * [Flexera community](https://community.flexera.com/s/)

  ### Insights from Gartner®

  Find a curated series of actionable and objective insights for IT executives and their teams. Get expert insights from valued analysts, courtesy of Flexera.

  [Discover More](https://www.flexera.com/resources/gartner-analyst-research)
* Resources
  + Column 1
    - [Resources](https://www.flexera.com/resources)
      * [Webinars](https://www.flexera.com/resources?type%5Bwebinar%5D=webinar)
      * [Videos](https://www.flexera.com/resources?type%5Bvideo%5D=video)
      * [Datasheets](https://www.flexera.com/resources?type%5Bdatasheet%5D=datasheet)
      * [White papers & reports](https://www.flexera.com/resources?type%5Bwhite-paper-industry-report%5D=white-paper-industry-report)
  + Column 2
    - Hide group
      * [Blog](/blog/)
      * [Case studies](https://www.flexera.com/resources/case-studies)
      * [Events](https://www.flexera.com/resources?type%5Bevent%5D=event)
      * [Analyst Research](https://www.flexera.com/resources/gartner-analyst-research)
      * [Glossary](https://www.flexera.com/resources/glossary)
      * [Demos & trials](https://www.flexera.com/resources?type%5Bdemo-trials%5D=demo-trials)
      * [Business value calculator](https://www.flexera.com/resources/business-value-calculator)

  ### Flexera 2025 IT Priorities Report

  Insights from Flexera’s 2025 IT Priorities Report highlight what’s top of mind for IT decision makers in the year ahead. Discover the challenges, priorities and opportunities that will shape the future IT landscape.

  [View Report](https://info.flexera.com/ITV-REPORT-IT-Priorities)
* About
  + Column 1
    - [Company](https://www.flexera.com/about-us)
      * [About](https://www.flexera.com/about-us)
      * [Careers](https://www.flexera.com/about-us/careers)
      * [Contact](https://www.flexera.com/about-us/contact-us)
      * [Leadership](https://www.flexera.com/about-us/leadership)
    - [Partners](https://www.flexera.com/about-us/partners)
      * [Partner program](https://www.flexera.com/about-us/partners/partner-program)
      * [Partner directory](https://www.flexera.com/about-us/partners/directory)
  + Column 2
    - [Press center](https://www.flexera.com/about-us/press-center)
      * [Press releases](https://www.flexera.com/about-us/all-press-releases)
      * [Awards](https://www.flexera.com/about-us/press-center#awards)
      * [Articles](https://www.flexera.com/about-us/all-articles)
    - Hide group
      * Social responsibility
        + [ESG](https://www.flexera.com/about-us/environmental-social-governance)
        + [Diversity](https://www.flexera.com/about-us/diversity)

  ### More value with technology intelligence

  The unparalleled synergy of Flexera and Snow provides the Technology Intelligence you need for more efficiency, insight and governance than ever before.

  [Discover More](https://www.flexera.com/more-value-with-technology-intelligence)

Search

en

* [English](https://www.flexera.com/products/security/software-vulnerability-research/secunia-research?referrer=secunia)
* [Deutsch](https://www.flexera.de/products/security/software-vulnerability-research/secunia-research?referrer=secunia)

## External Links

* External Links
  + [Community](https://community.flexera.com/)
  + [Product Access](https://app.flexera.com/login)
  + [Partner Portal](https://flexera.channeltivity.com/Login)

[Book a demo](/about-us/contact-us?C_Interest1=sales)

# Secunia Research

## The world’s best vulnerability intelligence

The Secunia Research team from Flexera provides the most accurate and reliable source of vulnerability intelligence.

[Contact Us](https://www.flexera.com/about-us/contact-us?C_Interest1=sales&C_SolutionInterest=SVM)
Watch video (0:29)

Related links

* [Anatomy of a security advisory](https://www.flexera.com/resources/infographics/anatomy-of-a-security-advisory)
* [Software Vulnerability Research](https://www.flexera.com/products/software-vulnerability-research)
* [Software Vulnerability Manager](/products/software-vulnerability-manager)
* [Security advisories from Secunia Research](https://www.flexera.com/products/security/software-vulnerability-advisories)
* [Report a vulnerability](https://www.flexera.com/about-us/contact-us/report-vulnerability)

 ![Secunia Research](/sites/default/files/2022-04/hero-secunia-research-bg.jpg)

Featured Details

## Multiple ways to consume Secunia Research

Secunia delivers software security research that provides reliable, curated and actionable vulnerability intelligence. Organizations can expect to receive standardized, validated and enriched vulnerability research on a specific version of a software product. Secunia Research supports four solutions:

![Software Vulnerability Research](/sites/default/files/2022-04/icon-secunia-research-svr.svg)

### [Software Vulnerability Research](https://www.flexera.com/products/software-vulnerability-research)

Software Vulnerability Research utilizes Secunia Research to drive awareness of vulnerabilities matching your specified criteria

[Learn More](https://www.flexera.com/products/software-vulnerability-research)

![Software Vulnerability Manager](/sites/default/files/2022-04/icon-secunia-research-svm.svg)

### [Software Vulnerability Manager](/products/software-vulnerability-manager)

Software Vulnerability Manager uses Secunia Research data to identify, prioritize and patch known vulnerable software detected in your environment

[Learn More](/products/software-vulnerability-manager)

![Data Platform](/sites/default/files/2022-04/icon-secunia-research-dp.svg)

### [Data Platform](https://www.flexera.com/products/data-platform)

Data Platform leverages Secunia Research to provide high-level insights based on major or minor versions of software in your normalized inventory

[Learn More](https://www.flexera.com/products/data-platform)

![Flexera One](/sites/default/files/2022-04/icon-secunia-research-flexera-one.svg)

### [Flexera One](/flexera-one)

Flexera One utilizes Secunia Research (alongside public NVD data) to provide more granular matching of build-level versions of software in your normalized inventory within its IT Asset Management and IT Visibility solutions

[Learn More](/flexera-one)

How it works

## Accurate, reliable vulnerability insights at your fingertips

The Secunia Research team from Flexera is comprised of several security specialists who conduct vulnerability research in various products in addition to testing, verifying and validating public vulnerability reports. Since its inception in 2002, the goal of the Secunia Research team is to provide the most accurate and reliable source of vulnerability intelligence.

Delivering the world’s best vulnerability intelligence requires skill and passion. Team members continually develop their skills exploring various high-profile closed and open-source software using a variety of approaches, focusing chiefly on thorough code audits and binary analysis. The team has received industry recognition, including naming members to [Microsoft’s Most Valuable Security Researchers](https://msrc-blog.microsoft.com/2019/08/07/announcing-2019-msrc-most-valuable-security-researchers/) list.

Secunia researchers discover hard-to-find vulnerabilities that aren’t normally identified with techniques such as fuzzing, and the results have been impressive. Members of the Secunia Research team have discovered critical vulnerabilities in products from vendors including Microsoft, Symantec, IBM, Adobe, RealNetworks, Trend Micro, HP, Blue Coat, Samba, CA, Mozilla and Apple.

The team produces invaluable security advisories based on research of the vulnerabilities affecting any given software update. Sometimes a single update can address multiple vulnerabilities of varying criticalities and threats; but these advisories aggregate and distill findings down to a single advisory perfect for the prioritization of patching efforts within [Software Vulnerability Manager](/products/software-vulnerability-manager). Criticality scores are consistently applied along with details around attack vector and other valuable details within [Software Vulnerability Research](/products/software-vulnerability-research/secunia-research). Illegitimate vulnerability reports are also investigated and rejected so you can focus only on what truly matters.

Informing IT, Transforming IT

## Industry insights to help keep you informed

[#### Webinar

### Stay Ahead of Cyber Threats: Flexera's Latest Vulnerability Insights

Join us for this session where we'll explore the latest findings from the Flexera Monthly Vulnerability Insights Report.](https://info.flexera.com/SVM-WBNR-Vulnerability-Insights-Roundtable)

[#### Webinar

### Dive deeper into the Flexera Annual Vulnerability Insights

We'll explore the key findings from the Flexera Annual Vulnerability Insights Report. Learn about the latest cybersecurity trends, the most targeted industries, the types of vulnerabilities, plus management and mitigation strategies.](https://info.flexera.com/SVM-WBNR-Flexera-Annual-Vulnerability-Insights?lead_source=Website%20Visitor&id=Flexera.com-Resources)

#### Video

### Close the Risk Window with Software Vulnerability Manager

Stop reacting. Gain control. Stay secure. Build a more effective risk mitigation process leveraging Secunia Research vulnerability intelligence and the largest repository of third-party patch data in the industry.

Remote video URL

[#### Trial

### Software Vulnerability Manager Assessment free trial

Get access to the complete set of modules of Software Vulnerability Manager: Research, Assessment and Patching](https://info.flexera.com/SVM-EVAL-Software-Vulnerability-Manager)

[#### Datasheet

### Protect your ServiceNow® investment with the highest quality data

IT Visibility offers certified ServiceNow integrations that accelerate platform expansion, improve ROI and increase efficiencies across ITIL processes by delivering clean software and hardware asset data directly.](/sites/default/files/datasheet-itv-maximize-servicenow-investment.pdf)

[#### Blog

### Avoid missing crucial vulnerability intelligence amid NVD backlog

Recent developments regarding the National Vulnerability Database (NVD) have some technology leaders on edge. Since February, the U.S. National Institute of Standards and Technology (NIST) has almost completely stopped enriching software vulnerabi...](https://www.flexera.com/blog/vulnerability-management/avoid-missing-crucial-vulnerability-intelligence-amid-nvd-backlog/)

[View all resources](https://www.flexera.com/resources?category%5Bsoftware-vulnerability-management%5D=software-vulnerability-management)

## Footer Menu

* Column
  + Business challenge
    - [Software renewals and audits](https://www.flexera.com/solutions/software-renewals-audits)
    - [Software license management and optimization](https://www.flexera.com/solutions/software-usage-costs)
    - [SaaS spend management](https://www.flexera.com/solutions/saas-spend)
    - [Cloud cost management](https://www.flexera.com/solutions/cloud-cost)
    - [IT asset lifecycle management](https://www.flexera.com/solutions/it-asset-lifecycle)
    - [CMDB data quality](https://www.flexera.com/solutions/cmdb-data-quality)
    - [Accurate IT inventory](https://www.flexera.com/solutions/it-inventory)
    - [Security and regulatory risk management](https://www.flexera.com/solutions/it-security-regulatory-risk)
    - [Sustainable IT](https://www.flexera.com/solutions/sustainable-it)
    - [AI-powered transformation](https://www.flexera.com/solutions/ai-powered-transformation)
    - [Public sector](https://www.flexera.com/solutions/public-sector)
* Column
  + Spend management by vendor
    - [IBM](https://www.flexera.com/solutions/vendor/ibm)
    - [Oracle](https://www.flexera.com/solutions/vendor/oracle)
    - [Microsoft](https://www.flexera.com/solutions/vendor/microsoft)
    - [SAP](https://www.flexera.com/solutions/vendor/sap)
    - [VMware](https://www.flexera.com/solutions/vendor/vmware)
    - [ServiceNow](https://www.flexera.com/solutions/vendor/servicenow)
    - [AWS](https://www.flexera.com/solutions/vendor/aws)
    - [Salesforce](https://www.flexera.com/solutions/vendor/salesforce)
    - [BMC](https://www.flexera.com/solutions/cmdb-data-quality/bmc)
    - [Adobe](https://www.flexera.com/solutions/vendor/adobe)
* Column
  + Products
    - [Flexera One](https://www.flexera.com/products/flexera-one)
    - [Snow Atlas](https://www.flexera.com/products/snow-atlas)
    - [Security](https://www.flexera.com/products/security)
    - [Application Readiness](https://www.flexera.com/products/adminstudio)
    - [All products](https://www.flexera.com/products)
    - [All Snow products](https://www.flexera.com/products/snow)
    - [Integrations](https://www.flexera.com/products/integrations)
* Column
  + Company
    - [About](https://www.flexera.com/about-us)
    - [Careers](https://www.flexera.com/about-us/careers)
    - [Leadership](https://www.flexera.com/about-us/leadership)
    - [Contact us](https://www.flexera.com/about-us/contact-us)
    - [Media / press center](https://www.flexera.com/about-us/press-center)
    - [Revenera.com](https://www.revenera.com)

 +1.800.374.4353

en

* [English](https://www.flexera.com/products/security/software-vulnerability-research/secunia-research?referrer=secunia)
* [Deutsch](https://www.flexera.de/products/security/software-vulnerability-research/secunia-research?referrer=secunia)

 [![Home](/themes/custom/flexera/images/logo.svg)](https://www.flexera.com/)

© 2025 Flexera. All Rights Reserved.

## Footer

* [Privacy Policy](https://www.flexera.com/legal/privacy-policy)
* [Terms and conditions](https://www.flexera.com/legal)
* [Contact Us](https://www.flexera.com/about-us/contact-us)
* [Impressum](https://www.flexera.com/about-us/impressum)
* [Site Map](https://www.flexera.com/sitemap)

#####

×

...



=== Content from gmod.org_85319439_20250124_223457.html ===

# [GMOD](https://gmod.org/)

# GMOD News

[Tweets by gmodproject](https://twitter.com/gmodproject)

this is a snapshot of the news page from right before MW was shutdown.

News that is relevant to the GMOD user and developer communities. This
page only shows recent news; to see all news items since autumn 2007, go
to the **[GMOD News
Archives](GMOD_News_Archives "GMOD News Archives")**. See also the [GMOD
Calendar and Events List](Calendar "Calendar").

## Contents

* [1 Recent
  News](#Recent_News)
* [2 News
  Archives](#News_Archives)
* [3 Adding a News
  Item](#Adding_a_News_Item)
  + [3.1 Short
    Instructions](#Short_Instructions)
  + [3.2 Longer
    Instructions](#Longer_Instructions)

## Recent News

| [Prospecting for Proposals for GSoC 2024](News/Prospecting_for_Proposals_for_GSoC_2024)[<img alt="GoogleSummer 2016logo.jpg" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/5/5d/GoogleSummer\_2016logo.jpg/272px-GoogleSummer\_2016logo.jpg.png" width="272" height="272" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/5/5d/GoogleSummer\_2016logo.jpg 1.5x, /mediawiki/images/5/5d/GoogleSummer\_2016logo.jpg 2x" />](GSoC "GSoC") The Open Genome Informatics group and GMOD has submitted an application for [Google Summer of Code (GSoC)](https://summerofcode.withgoogle.com/) this year, and we are soliciting project ideas from people and groups involved in the GMOD project.  GSoC is a global program run by Google to encourage students to get real-world experience of participating in a software project; students work on a project over the summer months and receive a stipend from Google for participating. Thousands of organizations of different sizes participate in GSoC, with the common factor being that they must produce open-source code.  Please take a look at the [GSoC wiki page](GSoC "GSoC") and think about any projects that might work well for a GSoC student, and add them to the wiki. If you are interested in being a mentor, don't hesitate to get in touch with us at robin.haw@oicr.on.ca help@gmod.org.  If you have any questions--e.g. what constitutes an appropriate project; whether your idea is sufficiently GMOD-related--please feel free to email rhaw@oicr.on.ca and help@gmod.org for advice!   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2024/02/12*[JBrowse2 v1.1.0 Released](News/JBrowse2_v1.1.0_Released)Contents  * 1 We're   pleased to announce a new release of JBrowse Web!   + 1.1 Changed callbacks language from JavaScript to     Jexl   + 1.2 Fetch     intron and upstream/downstream sequences   + 1.3 Interactive documentation using     Storybook   + 1.4 Enhanced navigation to drawer widget     stack   + 1.5 Downloads   + 1.6 1.1.0     (2021-03-29)     - 1.6.1 🚀       Enhancement     - 1.6.2 🐛       Bug Fix     - 1.6.3 📝       Documentation     - 1.6.4 🏠       Internal     - 1.6.5 Committers: 6  We're pleased to announce a new release of JBrowse Web! (Reposted by permission from <https://jbrowse.org/jb2/blog/2021/03/29/v1.1.0-release> Changed callbacks language from JavaScript to Jexl To allow users to safely and seamlessly share advanced configurations in sessions, we now use Jexl to express configuration callbacks. Note that this is a breaking change, function()-style callbacks will no longer work.  For details, see the [callbacks section of our configuration guide](https://jbrowse.org/jb2/docs/config_guide#configuration-callbacks). Fetch intron and upstream/downstream sequences We also have several other improvements including the ability to get intron and upstream/downstream sequence in the feature details  [<img alt="Upstream downstream.png" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/7/76/Upstream\_downstream.png/700px-Upstream\_downstream.png" width="700" height="248" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/7/76/Upstream\_downstream.png/1050px-Upstream\_downstream.png 1.5x, /mediawiki/images/thumb/7/76/Upstream\_downstream.png/1400px-Upstream\_downstream.png 2x" />](file:///Upstream_downstream.png) Interactive documentation using Storybook Another new update is the first release of our interactive Storybook docs for the embeddable React Linear Genome View. The docs contain live examples of how the LGV component can be used, along with source-code examples. The site can be found [here](https://jbrowse.org/storybook/lgv/v1.1.0). Enhanced navigation to drawer widget stack We have added a dropdown to enhance navigation between stack of active widgets. The update also adds a minimize button to allow quick access to full screen JBrowse web.  See below for demos of the new navigation UI.  [<img alt="Minimize button demo.gif" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/0/0f/Minimize\_button\_demo.gif" width="480" height="414" />](file:///Minimize_button_demo.gif)  Demo of using the minimize button in the drawer Downloads  * [v1.1.0](https://github.com/GMOD/jbrowse-components/releases/tag/v1.1.0)   To install JBrowse 2 for the web, you can download the link above, or you can use the JBrowse CLI to automatically download the latest version. See the JBrowse web quick start for more details. 1.1.0 (2021-03-29)🚀 Enhancement core   1. 1846 Improve copy+paste in the data grids for feature details    (@cmdcolin) 2. 1814 Add ability to get promoter sequence and intron sequence for    genes from the feature details panel (@cmdcolin) 3. 1816 Remove some animation effects (@cmdcolin) 4. 1778 Adds dropdown to show drawer widget stack (@teresam856) 5. 1685 Change callbacks language from JavaScript to Jexl    (@peterkxie)   Other   1. 1831 Add dialog for launching breakpoint split view from variant    feature details (@cmdcolin) 2. 1803 Transcript and gene glyphs can now display implied UTRs, active    by default (@cmdcolin) 3. 1808 Add another heuristic for returning gene features from BigBed    (@cmdcolin) 4. 1774 Add warning dialog in LGV before returning to import form to    prevent accidentally losing the current view (@cmdcolin)  🐛 Bug Fix core   1. 1811 Check for existence of window more robustly to allow in SSR or    node applications (@elliothershberg) 2. 1793 Fix dotplot rendering outside it's allowed bounds    (@cmdcolin) 3. 1783 Add hic aborting and fix remoteAbort signal propagation    (@cmdcolin) 4. 1723 A few bugfixes (@garrettjstevens)   Other   1. 1815 Clear tracks when using "Return to import form"    (@cmdcolin) 2. 1819 Standardized sentence casing on drawer widget titles    (@cmdcolin) 3. 1796 Bump generic-filehandle for fixing CORS errors from Chrome    cache pollution (@cmdcolin)  📝 Documentation  1. 1824 Add storybook docs page for nextjs usage    (@elliothershberg) 2. 1770 1469 storybook deploy (@elliothershberg) 3. 1807 Update developer guide to cover displays, and highlight working    external plugins (@cmdcolin) 4. 1779 Collaborative release announcement editing (@rbuels) 5. 1791 Add a couple more demos for our live version with MDX    (@cmdcolin)  🏠 Internal Other   1. 1820 Create v1.1.0.md, draft of release announcements    (@garrettjstevens) 2. 1823 Add note about previewing changelog to CONTRIBUTING.md    (@garrettjstevens)   core   1. 1834 Change jbrowse-components monorepo default branch from 'master'    to 'main' (@rbuels)  Committers: 6  * Colin Diesh (@cmdcolin) * Elliot Hershberg (@elliothershberg) * Garrett Stevens (@garrettjstevens) * Peter Xie (@peterkxie) * Robert Buels (@rbuels) * Teresa Martinez (@teresam856)    *Posted to the [GMOD News](GMOD_News "GMOD News") on 2021/03/30*[Prospecting for Proposals for GSoC 2021](News/Prospecting_for_Proposals_for_GSoC_2021)[<img alt="GoogleSummer 2016logo.jpg" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/5/5d/GoogleSummer\_2016logo.jpg/272px-GoogleSummer\_2016logo.jpg.png" width="272" height="272" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/5/5d/GoogleSummer\_2016logo.jpg 1.5x, /mediawiki/images/5/5d/GoogleSummer\_2016logo.jpg 2x" />](GSoC "GSoC") The Genome Informatics group and GMOD will be submitting an application for [Google Summer of Code (GSoC)](https://summerofcode.withgoogle.com/) this year, and we are soliciting project ideas from people and groups involved in the GMOD project.  GSoC is a global program run by Google to encourage students to get real-world experience of participating in a software project; students work on a project over the summer months and receive a stipend from Google for participating. Thousands of organizations of many different sizes take part in GSoC, with the common factor being that they must produce open-source code.  Please take a look at the [GSoC wiki page](GSoC "GSoC") and think about any projects that might work well for a GSoC student, and add them to the wiki. If you are interested in being a mentor, please contact us at robin.haw@oicr.on.ca help@gmod.org.  If you have any questions--e.g. what constitutes an appropriate project; whether your idea is sufficiently GMOD-related--please feel free to email robin.haw@oicr.on.ca and help@gmod.org for advice!   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2021/02/16*[Codefest 2020](Codefest_2020)[<img alt="Palms.jpg" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/d/df/Palms.jpg" width="1052" height="218" />](file:///Palms.jpg)   [Handlery Hotel in the Balboa room](https://sd.handlery.com/) (down the street from the Town and Country Hotel) San Diego, California January 9-10, 2020 </strong>  Immediately before [Plant and Animal Genome XXVIII (PAG 2020)](https://www.intlpag.org/2020/)  </center>  **[Participant list now available](https://docs.google.com/spreadsheets/d/1Kp-fGI0U88PtsbryP1j-BUGA4cduYvC2ojW9Mk0jwxU/edit#gid=0)**    There will be a GMOD codefest occurring before the Plant and Animal Genomes meeting in San Diego. The codefest will be at the Town and Country hotel on January 9 and 10. If you would like to suggest a problem or project to address, add it to this [Google Doc](https://docs.google.com/document/d/1_CnUW_W4tNyl7lSlihCwZDKT45VQQxcI3I-VgjnC2Dc/edit?usp=sharing). The codefest is open to anyone who'd like to work on any GMOD project (or, better yet, any combination of GMOD projects), including but not limited to [<img alt="GMOD Hackathon" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/7/7f/EvoHackLaptops2010.jpg/200px-EvoHackLaptops2010.jpg" width="200" height="119" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/7/7f/EvoHackLaptops2010.jpg/300px-EvoHackLaptops2010.jpg 1.5xhttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/7/7f/EvoHackLaptops2010.jpg 2x" />](file:///EvoHackLaptops2010.jpg "GMOD Hackathon")  * [Tripal](Tripal "Tripal") * [Chado](Chado "Chado") * [JBrowse](JBrowse "JBrowse") (1 and 2) * [Apollo](Apollo "Apollo") * [Galaxy](Galaxy "Galaxy") * [InterMine](InterMine "InterMine")   We already know that there will be Tripal, Chado, Apollo and JBrowse developers present. Agenda Thursday:   | 9:00-10:00 | Introduction and working group organization | | --- | --- | | 10:00-12:00 | Working groups | | 12:00-1:30 | Lunch (on your own) | | 1:30-4:00 | Working groups | | 4:00-5:00 | Report on progress |   Friday:   | 9:00-12:00 | Working groups | | --- | --- | | 12:00-1:30 | Lunch (on your own) | | 1:30-3:00 | Working groups | | 3:00-3:45 | Report on progress |  * Note: the Tripal steering committee will meet on Friday from 12:00 -   3:00 pm at the Postcard Bistro in the Handlery.    *Posted to the [GMOD News](GMOD_News "GMOD News") on 2020/01/08*[Tripal 3.5 Released](Tripal_3.5_Released) This [Tripal](http://tripal.info) release includes major performance improvements including an improved and blazing fast GFF3 loader. Follow standard Drupal updating procedures to update. Changes in this version:   1. Performance improvements to the Cross References, Relationships, and    References. 2. A rewritten GFF3 loader that is extremely fast, even with a fully    populated Tripal site. 3. A new Sequences fields that combines reference-dervied sequences    with primary sequences, and CDS/proteins for mRNA features and sequences    now have more informative definition lines for FASTA sequences. 4. Updates to the online documentation 5. Bug fixes in web services 6. Bug fixes in the OBO loader for ontologies that won't import.    *Posted to the [GMOD News](GMOD_News "GMOD News") on 2021/02/08*[JBrowse 2 release](News/JBrowse_2_release)[<img alt="JBrowse" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/5/5a/Jbrowse2\_logo.png/200px-Jbrowse2\_logo.png" width="200" height="67" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/5/5a/Jbrowse2\_logo.png/300px-Jbrowse2\_logo.png 1.5x, /mediawiki/images/thumb/5/5a/Jbrowse2\_logo.png/400px-Jbrowse2\_logo.png 2x" />](https://jbrowse.org/jb2/ "JBrowse") We are pleased to announce the first public release of [JBrowse 2](https://jbrowse.org/jb2/)!  JBrowse 2 is the successor to JBrowse, but is a completely new application written with modern web standards and frameworks.  Some of the features new to JBrowse 2 include:   * New types of views, including circular, synteny, and dotplot   views * Graphical configuration editing * Connections to resources such as UCSC Track Hubs   You can see some demos of JBrowse 2 in action [here](https://jbrowse.org/jb2/demos/). To get started using JBrowse 2, visit our quickstart guide [here](https://jbrowse.org/jb2/docs/quickstart_web/).  We'd love to hear what you think! You can find information about how to contact us [here](https://jbrowse.org/jb2/contact/).  The JBrowse Team *Posted to the [GMOD News](GMOD_News "GMOD News") on 2020/11/09*[Prospecting for Proposals for GSoC 2020](News/Prospecting_for_Proposals_for_GSoC_2020)[<img alt="GoogleSummer 2016logo.jpg" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/5/5d/GoogleSummer\_2016logo.jpg/272px-GoogleSummer\_2016logo.jpg.png" width="272" height="272" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/5/5d/GoogleSummer\_2016logo.jpg 1.5x, /mediawiki/images/5/5d/GoogleSummer\_2016logo.jpg 2x" />](GSoC "GSoC") The Genome Informatics group and GMOD will be submitting an application for [Google Summer of Code (GSoC)](https://summerofcode.withgoogle.com/) this year, and we are soliciting project ideas from people and groups involved in the GMOD project.  GSoC is a global program run by Google to encourage students to get real-world experience of participating in a software project; students work on a project over the summer months and receive a stipend from Google for participating. Thousands of organizations of many different sizes take part in GSoC, with the common factor being that they must produce open-source code.  Please take a look at the [GSoC wiki page](GSoC "GSoC") and think about any projects that might work well for a GSoC student, and add them to the wiki. If you are interested in being a mentor, please contact us at robin.haw@oicr.on.ca help@gmod.org.  If you have any questions--e.g. what constitutes an appropriate project; whether your idea is sufficiently GMOD-related--please feel free to email robin.haw@oicr.on.ca and help@gmod.org for advice!   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2020/02/03*[Call for PAG Abstracts](News/Call_for_PAG_Abstracts)Call for PAG Abstracts Time is short!  If you want to attend [PAG](http://www.intlpag.org/) and would like to present on a topic that would be of interest to the GMOD community, please send an abstract or at least a descriptive title to help@gmod.org. Types of talks typically include updates on GMOD software projects, usage stories for successful sites, proposals for new GMOD projects and descriptions of plugins for existing GMOD software projects like [Tripal](Tripal "Tripal"), [JBrowse](JBrowse "JBrowse") and [Galaxy](Galaxy "Galaxy").  Please consider giving a talk and sharing your experience and ideas!   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2016/11/04*[New GMOD Server](News/New_GMOD_Server)GMOD.org has a new home Due to a old server being retired, gmod.org has a new home. In the course of migrating the server, we also had to update the version of MediaWiki that is powering the site. If you notice any problems with gmod.org, please send an email to help at gmod dot org to let us know what's going on.   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2016/09/29*[GMOD-JBrowse 2016 Survey](News/GMOD-JBrowse_2016_Survey) Hello Genome Informaticians,  The following survey is aimed at users (and potential users) of GMOD genome databases, especially the JBrowse genome browser. It will directly inform the priorities for renewal of the R01 that funds JBrowse software development and the GMOD helpdesk.  We know surveys are thankless and dull. Your time in filling out this one is GREATLY appreciated.  <https://goo.gl/forms/1bKIuAMjGKrS0hUi1>  Thanks & best wishes,  The JBrowse team   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2016/09/23*[GCC2016](News/GCC2016)[<img alt="2016 Galaxy Community Conference" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1c/GCC2016Logo.png/250px-GCC2016Logo.png" width="250" height="167" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/1/1c/GCC2016Logo.png 1.5xhttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/1/1c/GCC2016Logo.png 2x" />](https://gcc2016.iu.edu/ "2016 Galaxy Community Conference") [<img alt="GMOD2016ColorsBigLetters 300px.png" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/e/e1/GMOD2016ColorsBigLetters\_300px.png/150px-GMOD2016ColorsBigLetters\_300px.png" width="150" height="126" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/e/e1/GMOD2016ColorsBigLetters\_300px.png/225px-GMOD2016ColorsBigLetters\_300px.png 1.5x, /mediawiki/images/thumb/e/e1/GMOD2016ColorsBigLetters\_300px.png/300px-GMOD2016ColorsBigLetters\_300px.png 2x" />](Jun_2016_GMOD_Meeting "Jun 2016 GMOD Meeting") **The [2016 Galaxy Community Conference (GCC2016)](https://gcc2016.iu.edu/) will be held June 25-29, at [Indiana University](https://gcc2016.iu.edu/location/) in Bloomington, Indiana, United states, immediately before the [June 2016 GMOD Meeting](Jun_2016_GMOD_Meeting "Jun 2016 GMOD Meeting"), also in Bloomington.** [Galaxy](Galaxy "Galaxy") is a GMOD Component which interacts with many other [GMOD Components](GMOD_Components "GMOD Components"), including:   * [Tripal](Tripal "Tripal"): A web front end for [Chado](Chado "Chado") databases. Galaxy is working   with the Tripal project to make Galaxy be Tripal's analysis engine. * [JBrowse](JBrowse "JBrowse"): A client-side genome browser and successor   to the venerable [GBrowse](GBrowse "GBrowse"). [JBrowse as a Galaxy Tool](http://gcc2015.tsl.ac.uk/Lightning/#JBrowse_as_a_Galaxy_Tool) was   presented by Helena Rasche at [GCC2015](http://gcc2015.tsl.ac.uk/). Ian Holmes, the   JBrowse PI, has put JBrowse-Galaxy integration at "top of the list" for   JBrowse's infrastructure upcoming infrastructure work. * [MAKER](MAKER "MAKER"): A genome annotation pipeline that integrates   several gene annotation engines, and combines them to produce annotation   that is better than any individual tool produces. A [MAKER-Galaxy](https://wiki.galaxyproject.org/Documents/Posters?action=AttachFile&do=view&target=ISMB2014_Kandalaft_GenomeAnnotation.pdf) by Agriculture and   Agri-Food Canada was presented at ISMB 2014. * [InterMine](InterMine "InterMine") and [BioMart](BioMart "BioMart"): These are both popular data sources that   are integrated with Galaxy.   Galaxy Community Conferences are an opportunity to participate in presentations, discussions, poster sessions, lightning talks and breakouts, all about high-throughput biology and the tools that support it.  The conference also includes [two days of training](https://gcc2016.iu.edu/training) offering in-depth topic coverage across several concurrent sessions, and [two days of hackathons](https://gcc2016.iu.edu/hacks/).  **[Oral presentation abstract submission](https://gcc2016.iu.edu/abstracts/) closes April 8; [poster and demo abstract submission](https://gcc2016.iu.edu/abstracts/) close May 20; and [scholarship applications](https://wiki.galaxyproject.org/News/GCC2016Scholarships) close May 1.**   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2016/04/04*[2016 GMOD Meeting](News/2016_GMOD_Meeting)[<img alt="GMOD2016ColorsBigLetters 300px.png" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/e/e1/GMOD2016ColorsBigLetters\_300px.png/260px-GMOD2016ColorsBigLetters\_300px.png" width="260" height="218" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/e/e1/GMOD2016ColorsBigLetters\_300px.png 1.5x, /mediawiki/images/e/e1/GMOD2016ColorsBigLetters\_300px.png 2x" />](Jun_2016_GMOD_Meeting "Jun 2016 GMOD Meeting") **The [next GMOD Community Meeting](Jun_2016_GMOD_Meeting "Jun 2016 GMOD Meeting") will be held at Indiana University in Bloomington, Indiana, United States, June 30-July 1, directly after the [2016 Galaxy Community Conference (GCC2016)](https://gcc2016.iu.edu/)**. [GMOD Meetings](Meetings "Meetings") are a mix of [user](GMOD_Membership#Users "GMOD Membership") and [developer](GMOD_Membership#Developers "GMOD Membership") presentations, and are a great place to find out what is happening in the project, what's coming up, and what others are doing.    **[Please register online at Eventbrite by June 20th 2016](https://gmod2016.eventbrite.com).** Early bird registration ends May 21.  For those who would like to present a talk or poster, the meeting registration form includes a section for submitting the presentation title and abstract.  If you have any suggestions or requests for the meeting, please contact the GMOD help desk.   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2016/04/04*[Please Support EcoCyc](News/Please_Support_EcoCyc)[<img alt="EcoCyc website" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/4/44/BioCyc.gif" width="216" height="60" />](http://ecocyc.org) [EcoCyc, the *E. coli* information resource](http://ecocyc.org) and one of the resources offered by the [Pathway Tools](Pathway_Tools "Pathway Tools") group, is in need of letters of support from the community after receiving a poor grant review, which could result in a complete loss of funding on July 1st, 2014. If you are a user of EcoCyc, please consider [writing a short letter in support of this vital resource](http://brg.ai.sri.com/ptools/ecocyc-letters-of-support.shtml). The deadline for letters is **May 26th, 2014**.  From the [Pathway Tools website](http://brg.ai.sri.com/ptools/ecocyc-letters-of-support.shtml)  **ECOCYC FUNDING CRISIS -- DEADLINE MAY 26**  EcoCyc received a very unfavorable grant review in February 2014. We are in discussions with the NIH to resolve this situation.  EcoCyc's usage has steadily increased. We made very strong progress on our challenging aims from the current grant period, and the project has produced many publications. EcoCyc received excellent reviews on previous grant applications. Furthermore, the needs of the prokaryotic research community for the content and software tools offered by EcoCyc have never been higher.  In the worst case, we will lose all funding on July 1, 2014 and be forced to re-apply. Even in the best case, we may receive a crippling funding cut that causes us to fall behind in its manual literature curation effort, and requires us to lay off experienced curation staff until funding can be obtained.  These events could seriously undermine EcoCyc, end the project altogether, or force us to begin charging usage fees.  To maintain EcoCyc as the free, up to date, and high-quality resource that you depend on, please tell the NIH what EcoCyc means to your research. Please click the button below to submit a PDF letter of support on institutional letterhead, or a short support statement, explaining the importance of EcoCyc.  We ask all regular users to submit; a short statement will take less than two minutes of your time. Students and post-docs, please ask your lab head to submit in addition to your submission.  More information on what to write and where to submit letters is available at [the Pathway Tools website](http://brg.ai.sri.com/ptools/ecocyc-letters-of-support.shtml).   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2014/05/21*[June 2014 WebApollo Hackathon](News/June_2014_WebApollo_Hackathon) Berkeley Bioinformatics Open-Source Projects (BBOP) invites you to join us this summer for a **[WebApollo Hackathon](http://genomearchitect.org/developers_hackathon_2014)** at [Lawrence Berkeley National Laboratory, California](http://lbl.gov).  **When**: Monday, June 2 - 6, 2014  **Where**: [Lawrence Berkeley National Laboratory. Building 74 (B74)](http://www.lbl.gov/LBL-Work/lab-site-map.html), Room 104. 1 Cyclotron Rd, Berkeley, CA 94720  **What**: five days of intensive, collaborative [WebApollo](WebApollo "WebApollo") development!  For five days developers will work on new features of interest to their research communities, improve existing features, and collaborate on developing features of interest to other colleagues.  Participants should:   * be able to set up their own WebApollo server before arriving in   Berkeley, including Postgres, Tomcat, etc. * code comfortably in JavaScript (client) and Java (server).   **What will we do while we are there?** You tell us! Send your suggestions for feature development, your questions, and any additional comments to apollo [dash] dev [at] lists [dot] lbl [dot] gov  Registration is free of charge, but tickets are required. [Register online!](https://www.eventbrite.com/e/web-apollo-developers-hackathon-tickets-11263235641)  Participants are responsible for arranging travel and accommodation on their own.  More details: [WebApollo Hackathon information](http://genomearchitect.org/developers_hackathon_2014)  *via Monica Munoz-Torres*   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2014-04-22*[Applications Open for GMOD Online Training](News/Applications_Open_for_GMOD_Online_Training) GMOD will be holding its first online training course for those interested in the set up and use of GMOD components.  The course will be held from **Monday 19th May to Friday 23rd May 2014**, and will cover core GMOD software, including [GBrowse](GBrowse "GBrowse") and [JBrowse](JBrowse "JBrowse"), [Galaxy](Galaxy "Galaxy"), [MAKER](MAKER "MAKER"), [Tripal](Tripal "Tripal"), [WebApollo](WebApollo "WebApollo"), [Canto](Canto "Canto"), and the [Chado](Chado "Chado") database.  If you are interested in attending, please see [GMOD Online Training 2014](GMOD_Online_Training_2014 "GMOD Online Training 2014") for more information and to submit your application.   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2014-03-31*[Canto Workshop at Biocuration 2014](News/Canto_Workshop_at_Biocuration_2014)[<img alt="CantoTextLogo.png" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/7/7e/CantoTextLogo.png/240px-CantoTextLogo.png" width="240" height="59" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/7/7e/CantoTextLogo.png/360px-CantoTextLogo.png 1.5x, /mediawiki/images/thumb/7/7e/CantoTextLogo.png/480px-CantoTextLogo.png 2x" />](Canto "Canto") [<img alt="ISBLogo.jpg" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/8/8d/ISBLogo.jpg" width="240" height="71" />](http://biocuration2014.events.oicr.on.ca/) GMOD will be running a workshop at [Biocuration 2014](http://biocuration2014.events.oicr.on.ca/) to demonstrate the use of [Canto](Canto "Canto"), on Wednesday 9th April in the afternoon. Canto is a literature curation tool that allows users to create functional annotations for genes and gene products using [OBO (ontology) terms](http://www.obofoundry.org/). Canto will soon be added to [GMOD in the Cloud](Cloud "Cloud"), and this workshop will show participants how to get a GMOD in the Cloud instance up and running--it takes less than ten minutes!--and how to use Canto for literature curation.  We will have more information closer to the time.   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2014/03/25*[Tripal 2.0a released](News/Tripal_2.0a_released) The Tripal Development Team is pleased to announce an alpha release of Tripal 2.0 for Drupal 7. This release is expected to have bugs and there is some functionality still under development. However, this release is made to help early adopters of Tripal for Drupal 7. Reports of bugs or other issues are highly welcomed. Below are available resources for Tripal 2.0a  ```    Download Instructions    New Functionality    Installation and Tutorial    Tripal 2.0 API site    Tripal Mailing List    Tripal Developer's Mailing List ```  The Installation tutorial is still under development but should have enough information for complete installation of Tripal v2.0a as well as loading of organisms and features.   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2014/02/27*[Precompiled Ontologies in Chado](News/Precompiled_Ontologies_in_Chado) Helena Rasche and the [Center for Phage Technology at Texas A&M University](https://cpt.tamu.edu/) are making [Chado database dumps of precompiled ontologies publicly available](https://cpt.tamu.edu/~docs/chado/) to save other Chado users the time and hassle of downloading and compiling the ontologies themselves.  Go to the [download site](https://cpt.tamu.edu/~docs/chado/).  The ontologies currently available are:   * Chado Feature Properties * Gene Ontology * Plant Ontology * Relationship Ontology * Sequence Ontology   These are updated weekly; the workflow is as follows:   * clone Chado from SVN * build * load ontologies * dump database as SQL * upload to a publicly accessible webserver   Please contact Helena if you are interested in having other ontologies added to the dumps, other builds with different (sub)sets of ontologies, or archived copies of schemas over time.  Note that none of the Chado-related scripts are installed, and the GMOD conf files are not created in GMOD\_ROOT. For remote access (e.g., via Artemis), and tools that do not make use of GMOD\_ROOT locally, this is not a problem.   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2014/02/20*[GCC2014 Registration is Open](News/GCC2014_Registration_is_Open)[<img alt="2014 Galaxy Community Conference (GCC2014)" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/a/a4/GCC2014Logo.png/300px-GCC2014Logo.png" width="300" height="126" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/a/a4/GCC2014Logo.png/450px-GCC2014Logo.png 1.5xhttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/a/a4/GCC2014Logo.png/600px-GCC2014Logo.png 2x" />](http://wiki.galaxyproject.org/Events/GCC2014 "2014 Galaxy Community Conference (GCC2014)") [<img alt="GCC2014 Training Day" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/4/43/GCC2014TDLogo.png/80px-GCC2014TDLogo.png" width="80" height="120" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/4/43/GCC2014TDLogo.png/120px-GCC2014TDLogo.png 1.5x, /mediawiki/images/thumb/4/43/GCC2014TDLogo.png/160px-GCC2014TDLogo.png 2x" />](http://wiki.galaxyproject.org/Events/GCC2014/TrainingDay "GCC2014 Training Day") **We are pleased to announce that [Early Registration](https://wiki.galaxyproject.org/Events/GCC2014/Register) and [Talk and Poster Abstract Submission](https://wiki.galaxyproject.org/Events/GCC2014/Abstracts) are now open for the [2014 Galaxy Community Conference (GCC2014)](https://wiki.galaxyproject.org/Events/GCC2014).**  [GCC2014](https://wiki.galaxyproject.org/Events/GCC2014) will be held at the [Homewood Campus](http://webapps.jhu.edu/jhuniverse/information_about_hopkins/campuses/homewood_campus/) of [Johns Hopkins University](http://jhu.edu), in [Baltimore, Maryland](http://visitors.baltimorecity.gov/), United States, from June 30 through July 2, 2014. [GCC2014](https://wiki.galaxyproject.org/Events/GCC2014) starts with a [Training Day](https://wiki.galaxyproject.org/Events/GCC2014/TrainingDay) featuring *five* parallel tracks, each with three, two and half hour long workshops. There are 13 different topics spanning the full Galactic spectrum of topics. [Take a look!](https://wiki.galaxyproject.org/Events/GCC2014/TrainingDay)  **[Early registration](https://wiki.galaxyproject.org/Events/GCC2014/Register)** is now open. Register early and *avoid paying 70% more for regular registration costs.*  Early registration is very affordable, with combined registration ([Training Day](https://wiki.galaxyproject.org/Events/GCC2014/TrainingDay) + main meeting) starting at $140 for post-docs and students. Registration is capped this year at 250 participants, *and we expect to hit that limit*. Registering early assures you a place at the conference and also a spot in the Training Day workshops you want to attend.  You can also book affordable conference housing at the same time you register. See the conference [Logistics page](https://wiki.galaxyproject.org/Events/GCC2014/Logistics) for details on this and other housing options.  **[Abstract submission](https://wiki.galaxyproject.org/Events/GCC2014/Abstracts)** for both oral presentations and posters is also open.  Abstract submission for oral presentations closes April 4, while poster submission closes April 25. Poster authors will be notified of acceptance status within two weeks of submission, while presentation authors will be notified no later than May2.  *Please consider presenting your work. If you are dealing with big biological data, then this meeting wants to hear about your work.* [<img alt="GigaScience Journal" src="http://gmod.ohttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/c/c5/GigaScienceLogo250.png/120px-GigaScienceLogo250.png" width="120" height="39" srcsethttps://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/c/c5/GigaScienceLogo250.png/180px-GigaScienceLogo250.png 1.5x, /mediawiki/images/thumb/c/c5/GigaScienceLogo250.png/240px-GigaScienceLogo250.png 2x" />](http://www.gigasciencejournal.com/ "GigaScience Journal") The *[GigaScience](http://www.gigasciencejournal.com/)* "Galaxy: Data Intensive and Reproducible Research" series announced for the last conference has published its [first papers](http://www.gigasciencejournal.com/series/Galaxy), *and is continuing to take submissions for this year's meeting and beyond*. BGI is also continuing to cover the article processing charges until the end of the year, and for more information see their [latest update](http://blogs.biomedcentral.com/gigablog/2014/02/06/rewarding-reproducibility-first-papers-in-our-galaxy-series-utilizing-our-gigagalaxy-platform/).  Thanks, and hope to see you in Baltimore!  The http:[GCC2014 Organizing Committee](https://wiki.galaxyproject.org/Events/GCC2014/Organizers)   *Posted to the [GMOD News](GMOD_News "GMOD News") on 2014/02/14*[GMOD Paper Cuts, Feb 10th, 2014](News/GMOD_Paper_Cuts%2C_Feb_10th%2C_2014) \_\_NOFACTBOX\_\_  **GMOD Paper Cuts is a periodic selection of choice cuts from the scientific literature featuring interesting, exciting, or otherwise eye-catching GMOD-related work.**  If you would like a paper to appear in GMOD Paper Cuts, please email the details to the GMOD helpdesk. Ideally the paper should be in an open-access publication so that anyone can read it.  For more GMOD and GMOD-related papers, and to contribute your own GMOD-related publications, [join our Mendeley group](http://mnd.ly/WwRe8F).    **Finding the missing honey bee genes: lessons learned from a genome upgrade** [1]  The first generation of genome sequence assemblies and annotations have had a significant impact upon our understanding of the biology of the sequenced species, the phylogenetic relationships among species, the study of populations within and across species, and have informed the biology of humans. As only a few Metazoan genomes are approaching finished quality (human, mouse, fly and worm), there is room for improvement of most genome assemblies. The honey bee (Apis mellifera) genome, published in 2006, was noted for its bimodal GC content distribution that affected the quality of the assembly in some regions and for fewer genes in the initial gene set (OGSv1.0) compared to what would be expected based on other sequenced insect genomes.  Here, we report an improved [honey bee genome assembly (Amel\_4.5)](http://hymenopteragenome.org:8080/Amel_4.5/jbrowse/) with a new gene annotation set (OGSv3.2), and show that the honey bee genome contains a number of genes similar to that of other insect genomes, contrary to what was suggested in OGSv1.0. The new genome assembly is more contiguous and complete and the new gene set includes ~5000 more protein-coding genes, 50% more than previously reported. About 1/6 of the additional genes were due to improvements to the assembly, and the remaining were inferred based on new RNAseq and protein data.  Lessons learned from this genome upgrade have important implications for future genome sequencing projects. Furthermore, the improvements significantly enhance genomic resources for the honey bee, a key model for social behavior and essential to global ecology through pollination.  Interesting findings from the new assembly of the honey bee genome, including many more genes than were found in the initial assembly. The [Hymenoptera Genome Database](http://hymenopteragenome.org) uses numerous GMOD resources, including [MAKER](MAKER "MAKER") for automated genome annotation, [JBrowse](JBrowse "JBrowse") and [GBrowse](GBrowse "GBrowse") for sequence browsing, and [WebApollo](WebApollo "WebApollo") for community genome annotation.    **Highly Specific and Efficient CRISPR/Cas9-Catalyzed Homology-Directed Repair in Drosophila** [2]  We and others recently demonstrated that the readily programmable CRISPR/Cas9 system can be used to edit the Drosophila genome. However, most applications to date have relied on aberrant DNA repair to stochastically generate frame-shifting indels and adoption has been limited by a lack of tools for efficient identification of targeted events. Here we report optimized tools and techniques for expanded application of the CRISPR/Cas9 system in Drosophila through homology-directed repair (HDR) with double-stranded DNA (dsDNA) donor templates that facilitate complex genome engineering through the precise incorporation of large DNA sequences including screenable markers. Using these donors, we demonstrate the replacement of a gene with exogenous sequences and the generation of a conditional allele. To optimize efficiency and specificity, we generated transgenic flies that express Cas9 in the germline, and directly compared HDR and off-target cleavage rates of different approaches for delivering CRISPR components. We also investigated HDR efficiency in a mutant background previously demonstrated to bias DNA repair towards HDR. Finally, we developed a web-based tool that identifies CRISPR target sites and evaluates their potential for off-target cleavage using empirically rooted rules. Overall, we have found that injection of a dsDNA donor and guide RNA-encoding plasmids into vasa-Cas9 flies yields the highest efficiency HDR, and that target sites can be selected to avoid off-target mutations. Efficient and specific CRISPR/Cas9-mediated HDR opens the door to a broad array of complex genome modifications and greatly expands the utility of CRISPR technology for Drosophila research.  **CRISPR** is one of the most exciting recent technological advancements of the past couple of years. This paper reports new techniques and tools for using the CRISPR/Cas9 system for complex genome engineering. For more information, see the [flyCRISPR website](http://flycrispr.molbio.wisc.edu/).    **Analysis of Global Gene Expression in *Brachypodium distachyon* Reveals Extensive Network Plasticity in Response to Abiotic Stress** [3]  *Brachypodium distachyon* is a close relative of many important cereal crops. Abiotic stress tolerance has a significant impact on productivity of agriculturally important food and feedstock crops. Analysis of the transcriptome of Brachypodium after chilling, high-salinity, drought, and heat stresses revealed diverse differential expression of many transcripts. Weighted Gene Co-Expression Network Analysis revealed 22 distinct gene modules with specific profiles of expression under each stress. Promoter analysis implicated short DNA sequences directly upstream of module members in the regulation of 21 of 22 modules. Functional analysis of module members revealed enrichment in functional terms for 10 of 22 network modules. Analysis of condition-specific correlations between differentially expressed gene pairs revealed extensive plasticity in the expression relationships of gene pairs. Photosynthesis, cell cycle, and cell wall expression modules were down-regulated by all abiotic stresses. Modules which were up-regulated by each abiotic stress fell into diverse and unique gene ontology GO categories. This study provides genomics resources and improves our understanding of abiotic stress responses of Brachypodium.  Check out the JBrowse-powered [Brachypodium web genome browser](http://jbrowse.brachypodium.org) and other resources on the new [Brachypodium website](http://brachypodium.org)!    **Analyses of Hypomethylated Oil Palm Gene Space**[4]  Demand for palm oil has been increasing by an average of ~8% the past decade and currently accounts for about 59% of the world's vegetable oil market. This drives the need to increase palm oil production. Nevertheless, due to the increasing need for sustainable production, it is imperative to increase productivity rather than the area cultivated. Studies on the oil palm genome are essential to help identify genes or markers that are associated with important processes or traits, such as flowering, yield and disease resistance. To achieve this, 294,115 and 150,744 sequences from the hypomethylated or gene-rich regions of Elaeis guineensis and E. oleifera genome were sequenced and assembled into contigs. An additional 16,427 shot-gun sequences and 176 bacterial artificial chromosomes (BAC) were also generated to check the quality of libraries constructed. Comparison of these sequences revealed that although the methylation-filtered libraries were sequenced at low coverage, they still tagged at least 66% of the RefSeq supported genes in the BAC and had a filtration power of at least 2.0. A total 33,752 microsatellites and 40,820 high-quality single nucleotide polymorphism (SNP) markers were identified. These represent the most comprehensive collection of microsatellites and SNPs to date and would be an important resource for genetic mapping and association studies. The gene models predicted from the assembled contigs were mined for genes of interest, and 242, 65 and 14 oil palm transcription factors, resistance genes and miRNAs were identified respectively. Examples of the transcriptional factors tagged include those associated with floral development and tissue culture, such as homeodomain proteins, MADS, Squamosa and Apetala2. The E. guineensis and E. oleifera hypomethylated sequences provide an important resource to understand the molecular mechanisms associated with important agronomic traits in oil palm.  The newly-sequenced [oil palm genome](http://genomsawit.mpob.gov.my/genomsawit/) used the [MAKER](MAKER "MAKER") automated annotation pipeline. The oil palm is one of a number of genomics projects taking off in Malaysia at the moment. Perfect timing for a [GMOD workshop](GMOD_Malaysia_2014 "GMOD Malaysia 2014")!    **Production of a reference transcriptome and transcriptomic database ([EdwardsiellaBase](http://cnidarians.bu.edu/EdwardBase/cgi-bin/index.cgi)) for the lined sea anemone, *Edwardsiella lineata*, a parasitic cnidarian** [5]  The lined sea anemone *Edwardsiella lineata* is an informative model system for evolutionary-developmental studies of parasitism. In this species, it is possible to compare alternate developmental pathways leading from a larva to either a free-living polyp or a vermiform parasite that inhabits the mesoglea of a ctenophore host. Additionally, E. lineata is confamilial with the model cnidarian *Nematostella vectensis*, providing an opportunity for comparative genomic, molecular and organismal studies.  [...]  The transcriptomic data and database described here provide a platform for studying the evolutionary developmental genomics of a derived parasitic life cycle. In addition, these data from E. lineata will aid in the interpretation of evolutionary novelties in gene sequence or structure that have been reported for the model cnidarian *N. vectensis* (e.g., the split NF-κB locus). Finally, we include custom computational tools to facilitate the annotation of a transcriptome based on high-throughput sequencing data obtained from a “non-model system.”  Information and resources for the newly-sequenced cnidarian *E. lineata*; all genomic data is publicly available at [EdwardsiellaBase](http://cnidarians.bu.edu/EdwardBase/cgi-bin/index.cgi), and can be searched according to contig ID, gene ontology, protein family motif (Pfam), enzyme commission number, and BLAST. The alignment of the raw reads to the contigs can also be visualized via [JBrowse](JBrowse "JBrowse").    Happy reading!     1. ↑ **Cite error:    Invalid `<ref>` tag; no text was provided for refs    named `DOI:10.1186.2F1471-2164-15-86`** 2. ↑ **Cite error:    Invalid `<ref>` tag; no text was provided for refs    named    `DOI:10.1534.2Fgenetics.113.160713`** 3. ↑ **Cite error:    Invalid `<ref>` tag; no text was provided for refs    named    `DOI:10.1371.2Fjournal.pone.0087499`** 4. ↑ **Cite error:    Invalid `<ref>` tag; no text was provided for refs    named    `DOI:10.1371.2Fjournal.pone.0086728`** 5. ↑ **Cite error:    Invalid `<ref>` tag; no text was provided for refs    named `DOI:10.1186.2F1471-2164-15-71`**  Disclaimer: the papers included in this feature are for your entertainment and edification only. Inclusion does not imply an endorsement of the material or any association between the authors and the GMOD project.  *Posted to the [GMOD News](GMOD_News "GMOD News") on 2014/02/10* |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |

## [News Archives](GMOD_News_Archives "GMOD News Archives")

The [GMOD News Archives](GMOD_News_Archives "GMOD News Archives") lists
all news items since autumn 2007.

## Adding a News Item

**Note:** If you don’t want to do this yourself, please send the item to
the
GMOD
Help Desk and we will post it for you.

GMOD news items are wiki pages with the prefix “News/”. Creating a news
item will automatically add it to the RSS feed, which appears on the
**GMOD News** and [GMOD News
Archives](GMOD_News_Archives "GMOD News Archives") pages, and in the
news tracker on the [home page](Main_Page "Main Page").

### Short Instructions

Please follow these guidelines when adding a news item.

1. Create a new page named “**News/\***News Item Title\*”.
   * Please make *News Item Title* be as short as possible, **and no
     more 35 characters at most**.
2. Enter the text of your news item.
   * **The first link in the news item should point to the page/URL you
     want the RSS feed to link to.**
3. Preview / save your changes. Edit and save the page until the news
   item looks like you want.
4. Once you are happy with how the item looks, insert this line at the
   end of it:

```
 (don't do this--it was for the mediawiki instance)

```
### Longer Instructions

Please follow these guidelines when adding a news item.

1. Create a new page named “**News/\***News Item Title\*”.

   * Please make *News Item Title* be as short as possible, **and no
     more 35 characters at most**.
   * The page must start with **News/**, otherwise it won’t be picked
     by the RSS news feed.
2. Enter the text of your news item.

   * Make the news item succinct.
   * Do not start the item with a MediaWiki header (“== Header ==”).
     + It won’t render very well. You should be able to avoid headers
       altogether.
   * **The first link in the news item should point to the page/URL you
     want the RSS feed to link to.**
     + For example, the first link in a news item about
       [Chado](Chado "Chado")
       could point to the
       [Chado](Chado "Chado")
       page.
   * You can include images in your news item. See [preexisting news
     items](Category%253ANews_Items "Category%253ANews Items") for what markup
     to use to do this.
3. Preview / save your changes. Edit and save the page until the news
   item looks like you want.
4. Once you are happy with how the item looks, insert this line at the
   end of it:

   ```
   No--this was for MW

   ```

   Where *yyyy/mm/dd* is the current date. This line will cause it to
   be added to the GMOD News RSS feed. The item will show up in the
   feed within an hour.
5. **Once you have added the NewsItem template and saved the page, try
   to avoid editing the page after that.**

   * Every edit results in the news item being reposted to the RSS
     feed. If you do need to update an item later on, you may do so,
     but please also update the NewsItem line:
   ```
   NewsItem|yyyy/mm/dd, updated yyyy/mm/dd

   ```

[Category](Special%253ACategories "Special%253ACategories"):

* [GMOD Community](Category%253AGMOD_Community "Category%253AGMOD Community")

## Navigation menu

### Navigation

* [GMOD Home](Main_Page)
* [Software](GMOD_Components)
* [Categories /
  Tags](Categories)

### Documentation

* <Overview>
* [FAQs](Category%253AFAQ)
* [HOWTOs](Category%253AHOWTO)
* <Glossary>

### Community

* [GMOD News](GMOD_News)
* [Training /
  Outreach](Training_and_Outreach)
* <Support>
* [GMOD Promotion](GMOD_Promotion)
* <Meetings>
* <Calendar>

### Tools

* [Browse
  properties](Special%253ABrowse/GMOD_News)
* Last updated at 19:40 on 13 December 2018.
* Content is available under
  [a GNU Free Documentation License](http://www.gnu.org/licenses/fdl-1.3.html) unless otherwise
  noted.


